<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-05-22" />

<title>Assessing heritability enrichment for lung OCRs</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Lung_scMultiomics_paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/Lung_scMultiomics_paper">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Assessing heritability enrichment for lung
OCRs</h1>
<h4 class="date">2025-05-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Lung_scMultiomics_paper/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250512code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250512)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250512code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250512)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxinhelabLungscMultiomicspapertree2575a8afedb2abe7a9cf22fddd028d8041499727targetblank2575a8aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/tree/2575a8afedb2abe7a9cf22fddd028d8041499727" target="_blank">2575a8a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxinhelabLungscMultiomicspapertree2575a8afedb2abe7a9cf22fddd028d8041499727targetblank2575a8aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/tree/2575a8afedb2abe7a9cf22fddd028d8041499727" target="_blank">2575a8a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  ArchRLogs/
    Untracked:  Lung_scMultiomics_paper.Rproj
    Untracked:  _workflowr.yml
    Untracked:  analysis/ArchRLogs/
    Untracked:  analysis/about.knit.md
    Untracked:  analysis/archive.Rmd
    Untracked:  analysis/figures_for_grant_application.Rmd
    Untracked:  analysis/link_peaks_to_genes.Rmd
    Untracked:  code/run_GO_enrichment.R
    Untracked:  data/p2g_res/
    Untracked:  output/u19_multiomics

Unstaged changes:
    Modified:   README.md
    Modified:   analysis/identify_cell_types.Rmd
    Modified:   analysis/identify_lung_specific_transcriptomic_features.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/test.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/heritability_enrichment_for_lung_open_chromatin.Rmd</code>)
and HTML
(<code>docs/heritability_enrichment_for_lung_open_chromatin.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/2575a8afedb2abe7a9cf22fddd028d8041499727/analysis/heritability_enrichment_for_lung_open_chromatin.Rmd" target="_blank">2575a8a</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
<td>
figure for h2g enrichment
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xinhe-lab/Lung_scMultiomics_paper/9b45ce0b21aa5e220c54e9f9264ad311d80f8fa4/docs/heritability_enrichment_for_lung_open_chromatin.html" target="_blank">9b45ce0</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/25b622a05744a8c798c730760a597a93fa681fa1/analysis/heritability_enrichment_for_lung_open_chromatin.Rmd" target="_blank">25b622a</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
<td>
figure for h2g enrichment
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="compare-and-contrast-between-lung-and-blood"
class="section level1">
<h1>Compare and contrast between lung and blood</h1>
<div id="peak-overlapping" class="section level2">
<h2>Peak overlapping</h2>
<div id="barplots-to-show-peaks-shared-with-blood-or-not"
class="section level3">
<h3>Barplots to show peaks shared with blood or not</h3>
<p><img src="figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-2-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9b45ce0b21aa5e220c54e9f9264ad311d80f8fa4/docs/figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-2-1.png" target="_blank">9b45ce0</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="venn-diagram-to-compare-against-a-union-set-of-peaks"
class="section level3">
<h3>Venn Diagram to compare against a union set of peaks</h3>
<p>Lung and Blood peaks were called by different procedures. Lung peaks
have fixed size (500bp), while blood peaks have sizes ranging from 200
to 5K bp. To make comparison, I first made a union set of peaks and then
comparing peaks from each tissue against the union set.</p>
<p><img src="figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9b45ce0b21aa5e220c54e9f9264ad311d80f8fa4/docs/figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-3-1.png" target="_blank">9b45ce0</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="heritability-enrichment-with-s-ldsc" class="section level2">
<h2>heritability enrichment with S-LDSC</h2>
<div id="one-focal-cell-type-vs.-lung-and-blood-union-set"
class="section level3">
<h3>One focal cell-type vs. lung and blood union set</h3>
<p><img src="figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-9-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9b45ce0b21aa5e220c54e9f9264ad311d80f8fa4/docs/figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-9-1.png" target="_blank">9b45ce0</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Bubble plot</strong></p>
<p><img src="figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-11-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9b45ce0b21aa5e220c54e9f9264ad311d80f8fa4/docs/figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-11-1.png" target="_blank">9b45ce0</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="individual-test-across-lung-and-blood-major-lymphocytes"
class="section level3">
<h3>Individual test across lung and blood major lymphocytes</h3>
<p>For individual test, the quantity of heritability enrichment is used
to demonstrate overall contribution of each annotation. We can use
p-values for cell-type specific coefficients to compare their
contributions across cell types.</p>
<p>Legends:</p>
<ul>
<li>x-axis: heritability enrichment</li>
<li>p-values on top of the data points: p-values for tau*</li>
</ul>
<p>All lung and spleen lymphocytes show significant enrichment for
genetic risks of asthma and related diseases, but not for BMI and
height. B cells are less significant than T and NK cells.</p>
<p><img src="figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9b45ce0b21aa5e220c54e9f9264ad311d80f8fa4/docs/figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-12-1.png" target="_blank">9b45ce0</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="joint-test-of-lung-and-blood-separately-for-each-major-lymphocyte"
class="section level3">
<h3>Joint test of lung and blood separately for each major
lymphocyte</h3>
<p>Compared with blood, open chromatin regions of lung T cells are
significantly enriched for genetic risks of allergy, atopic dermatitis,
and asthma, but not for control traits as well as IPF and COPD. Lung NK
cells also show moderate enrichment.</p>
<p><img src="figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-13-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9b45ce0b21aa5e220c54e9f9264ad311d80f8fa4/docs/figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-13-1.png" target="_blank">9b45ce0</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="joint-test-of-lung-and-blood-t-subsets" class="section level3">
<h3>Joint test of lung and blood T subsets</h3>
<ul>
<li>Lung regulatory T cells show significant enrichment across all
asthma and related diseases</li>
<li>Lung CD4+ T, CD8+T and Th17 cells show enrichment for genetic risks
of specific traits.<br />
</li>
<li>Activating CD4+ T cells in blood consistently show enrichment for
risks of atopic dermatitis.</li>
</ul>
<pre><code>$RegT</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_text()`).</code></pre>
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
<p><img src="figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-14-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9b45ce0b21aa5e220c54e9f9264ad311d80f8fa4/docs/figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-14-1.png" target="_blank">9b45ce0</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$CD4T</code></pre>
<p><img src="figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-14-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9b45ce0b21aa5e220c54e9f9264ad311d80f8fa4/docs/figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-14-2.png" target="_blank">9b45ce0</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$CD8T</code></pre>
<p><img src="figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-14-3.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-3">
Past versions of unnamed-chunk-14-3.png
</button>
</p>
<div id="fig-unnamed-chunk-14-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9b45ce0b21aa5e220c54e9f9264ad311d80f8fa4/docs/figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-14-3.png" target="_blank">9b45ce0</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Th17</code></pre>
<p><img src="figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-14-4.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-4">
Past versions of unnamed-chunk-14-4.png
</button>
</p>
<div id="fig-unnamed-chunk-14-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9b45ce0b21aa5e220c54e9f9264ad311d80f8fa4/docs/figure/heritability_enrichment_for_lung_open_chromatin.Rmd/unnamed-chunk-14-4.png" target="_blank">9b45ce0</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="overlapping-cell-type-specific-peaks-with-coa-gwas-snps"
class="section level2">
<h2>Overlapping cell-type specific peaks with COA GWAS SNPs</h2>
<p><strong>Differential CA peaks from MAST</strong></p>
<p>We check differential CA peaks aggregated from different cell types
at each fdr threshold and plot quantile-quantile plots for GWAS SNPs
within those peaks. The LD blocks were labeled on SNPs at z-score
greater than 2.</p>
<p>`</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] cowplot_1.1.3                          
 [2] liftOver_1.22.0                        
 [3] Homo.sapiens_1.3.1                     
 [4] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
 [5] org.Hs.eg.db_3.16.0                    
 [6] GO.db_3.16.0                           
 [7] OrganismDbi_1.40.0                     
 [8] GenomicFeatures_1.50.4                 
 [9] AnnotationDbi_1.60.2                   
[10] rtracklayer_1.58.0                     
[11] gwascat_2.30.0                         
[12] scales_1.3.0                           
[13] colorRamp2_0.1.0                       
[14] eulerr_7.0.2                           
[15] rhdf5_2.42.1                           
[16] SummarizedExperiment_1.28.0            
[17] Biobase_2.58.0                         
[18] MatrixGenerics_1.10.0                  
[19] Rcpp_1.0.14                            
[20] Matrix_1.6-5                           
[21] GenomicRanges_1.50.2                   
[22] GenomeInfoDb_1.34.9                    
[23] IRanges_2.32.0                         
[24] S4Vectors_0.36.2                       
[25] BiocGenerics_0.44.0                    
[26] matrixStats_1.5.0                      
[27] data.table_1.17.0                      
[28] stringr_1.5.1                          
[29] plyr_1.8.9                             
[30] magrittr_2.0.3                         
[31] ggplot2_3.5.2                          
[32] gtable_0.3.6                           
[33] gtools_3.9.5                           
[34] gridExtra_2.3                          
[35] ArchR_1.0.2                            
[36] tidyr_1.3.1                            
[37] dplyr_1.1.4                            

loaded via a namespace (and not attached):
 [1] colorspace_2.1-1         rjson_0.2.23             rprojroot_2.0.4         
 [4] XVector_0.38.0           fs_1.6.5                 rstudioapi_0.17.1       
 [7] farver_2.1.2             bit64_4.0.5              xml2_1.3.8              
[10] codetools_0.2-20         splines_4.2.0            snpStats_1.48.0         
[13] cachem_1.1.0             knitr_1.50               polyclip_1.10-7         
[16] jsonlite_2.0.0           workflowr_1.7.1          Rsamtools_2.14.0        
[19] dbplyr_2.5.0             png_0.1-8                graph_1.76.0            
[22] BiocManager_1.30.25      readr_2.1.5              compiler_4.2.0          
[25] httr_1.4.7               fastmap_1.2.0            cli_3.6.4               
[28] later_1.4.2              htmltools_0.5.8.1        prettyunits_1.2.0       
[31] tools_4.2.0              glue_1.8.0               GenomeInfoDbData_1.2.9  
[34] rappdirs_0.3.3           jquerylib_0.1.4          vctrs_0.6.5             
[37] Biostrings_2.66.0        rhdf5filters_1.10.1      polylabelr_0.3.0        
[40] xfun_0.52                lifecycle_1.0.4          restfulr_0.0.15         
[43] XML_3.99-0.18            zlibbioc_1.44.0          BSgenome_1.66.3         
[46] VariantAnnotation_1.44.1 hms_1.1.3                promises_1.3.2          
[49] RBGL_1.74.0              parallel_4.2.0           yaml_2.3.10             
[52] curl_6.2.2               memoise_2.0.1            sass_0.4.9              
[55] biomaRt_2.54.1           stringi_1.8.4            RSQLite_2.3.9           
[58] BiocIO_1.8.0             filelock_1.0.3           BiocParallel_1.32.6     
[61] rlang_1.1.5              pkgconfig_2.0.3          bitops_1.0-9            
[64] evaluate_1.0.3           lattice_0.22-7           purrr_1.0.4             
[67] Rhdf5lib_1.20.0          labeling_0.4.3           GenomicAlignments_1.34.1
[70] bit_4.6.0                tidyselect_1.2.1         R6_2.6.1                
[73] generics_0.1.3           DelayedArray_0.24.0      DBI_1.2.3               
[76] pillar_1.10.2            whisker_0.4.1            withr_3.0.2             
[79] survival_3.8-3           KEGGREST_1.38.0          RCurl_1.98-1.17         
[82] tibble_3.2.1             crayon_1.5.3             BiocFileCache_2.6.1     
[85] tzdb_0.5.0               rmarkdown_2.29           progress_1.2.3          
[88] blob_1.2.4               git2r_0.33.0             digest_0.6.37           
[91] httpuv_1.6.15            munsell_0.5.1            viridisLite_0.4.2       
[94] bslib_0.9.0             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
