<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-05-15" />

<title>Identify lung-specific transcriptomic features</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Lung_scMultiomics_paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/Lung_scMultiomics_paper">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Identify lung-specific transcriptomic
features</h1>
<h4 class="date">2025-05-15</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Lung_scMultiomics_paper/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250512code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250512)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250512code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250512)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxinhelabLungscMultiomicspapertree3cae8973843a77e4fde9a85b6cb30fc603f55b20targetblank3cae897a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/tree/3cae8973843a77e4fde9a85b6cb30fc603f55b20" target="_blank">3cae897</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxinhelabLungscMultiomicspapertree3cae8973843a77e4fde9a85b6cb30fc603f55b20targetblank3cae897a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/tree/3cae8973843a77e4fde9a85b6cb30fc603f55b20" target="_blank">3cae897</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  ArchRLogs/
    Untracked:  Lung_scMultiomics_paper.Rproj
    Untracked:  _workflowr.yml
    Untracked:  analysis/ArchRLogs/
    Untracked:  analysis/about.knit.md
    Untracked:  analysis/archive.Rmd
    Untracked:  output/u19_multiomics

Unstaged changes:
    Modified:   README.md
    Modified:   analysis/identify_cell_types.Rmd
    Modified:   analysis/test.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/identify_lung_specific_transcriptomic_features.Rmd</code>)
and HTML
(<code>docs/identify_lung_specific_transcriptomic_features.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/3cae8973843a77e4fde9a85b6cb30fc603f55b20/analysis/identify_lung_specific_transcriptomic_features.Rmd" target="_blank">3cae897</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-15
</td>
<td>
added comments
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xinhe-lab/Lung_scMultiomics_paper/53899adf81c649ffd490b78d635e71a2a0d3603d/docs/identify_lung_specific_transcriptomic_features.html" target="_blank">53899ad</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/dab59d1d2c8ad6dd6a57dd0eeb48c74c36237c24/analysis/identify_lung_specific_transcriptomic_features.Rmd" target="_blank">dab59d1</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-15
</td>
<td>
updated DEG analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/1c96702c2460d67ec26402c2a80679e4becc833b/analysis/identify_lung_specific_transcriptomic_features.Rmd" target="_blank">1c96702</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
<td>
fixed errors for table
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xinhe-lab/Lung_scMultiomics_paper/497b42c341ba272e118cfec3270a9b21b27576f5/docs/identify_lung_specific_transcriptomic_features.html" target="_blank">497b42c</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/4faa63efe67cabebdf3ea8f6ada5f5ee67c3fa44/analysis/identify_lung_specific_transcriptomic_features.Rmd" target="_blank">4faa63e</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
<td>
DEG analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xinhe-lab/Lung_scMultiomics_paper/9008bd6d61a5fd68cbc2639bec30281d654abacb/docs/identify_lung_specific_transcriptomic_features.html" target="_blank">9008bd6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/c40f0da66d1c349627657922366e98a1ce44c255/analysis/identify_lung_specific_transcriptomic_features.Rmd" target="_blank">c40f0da</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
<td>
DEG analyses
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="differential-gene-expression-analyses-across-tissue"
class="section level1">
<h1>Differential gene expression analyses across tissue</h1>
<p>Wilcoxon ranksum test at single-cell leveL gives more conservative
results.</p>
<div id="summarizing-de-genes-by-selecting-p-value-cutoffs"
class="section level2">
<h2>Summarizing DE genes by selecting p-value cutoffs</h2>
<p>A table of cell counts by tissue and cell-type.</p>
<pre><code>          
           lungs spleens
  Other     1654     104
  Treg      1336      47
  Th17      2732      68
  CD4_T     6980     886
  CD8_T    12210     421
  NK        8067     464
  Memory_B  5287   10507
  Naive_B   1174    1710</code></pre>
<p>A barplot for number of DE genes detected for each cell type except
for Th17 and Treg, due to low number of cells in spleen.</p>
<p><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/53899adf81c649ffd490b78d635e71a2a0d3603d/docs/figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-2-1.png" target="_blank">53899ad</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/497b42c341ba272e118cfec3270a9b21b27576f5/docs/figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-2-1.png" target="_blank">497b42c</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9008bd6d61a5fd68cbc2639bec30281d654abacb/docs/figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-2-1.png" target="_blank">9008bd6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="compare-and-contrast-de-genes-across-immune-subsets"
class="section level3">
<h3>Compare and contrast DE genes across immune subsets</h3>
<p>A Venn diagram for DE genes shared across cell types other than
memory B cells implies DE genes are cell-type specific.</p>
<p><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9008bd6d61a5fd68cbc2639bec30281d654abacb/docs/figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-3-1.png" target="_blank">9008bd6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>A full summary of shared and unique DE genes across cell
types</strong></p>
<p>UpSet-style plot only shows the count of elements specific to each
intersection.</p>
<p>Lung up-regulated genes</p>
<p><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9008bd6d61a5fd68cbc2639bec30281d654abacb/docs/figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-4-1.png" target="_blank">9008bd6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Check the lung up-regulated genes shared across all immune
subsets</p>
<pre><code>[1] &quot;HSP90AA1 HSPA1A HSP90AB1 HSPD1 RPS26 DNAJB1 HSPA6 HSPA1B HSPH1 HSPE1 HSPB1 CACYBP HSPA8 UBC DOK2 BAG3 STIP1 ABHD3 PLIN2 ZFAND2A GNLY FKBP4 TNFRSF1B GBP2 CGAS MT2A HSPA4 NKG7 AHSA1 SERPINH1 CCL4L2 GZMB GIPC1&quot;</code></pre>
<p>Spleen up-regulated genes
<img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9008bd6d61a5fd68cbc2639bec30281d654abacb/docs/figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-6-1.png" target="_blank">9008bd6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="go-enrichment-results" class="section level3">
<h3>GO enrichment results</h3>
<p><strong>Lung Up-regulated genes</strong></p>
<p>Overall, lung up-regulated genes are enriched for immune-related
hallmark gene sets and GO terms for T cell activation, response to
cytokines, and regulation of cell adhesion. Particularly, lung-specific
genes in CD4+T and memory B cells are enriched for Th1/Th2
differentiation.</p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9008bd6d61a5fd68cbc2639bec30281d654abacb/docs/figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-7-1.png" target="_blank">9008bd6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-7-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9008bd6d61a5fd68cbc2639bec30281d654abacb/docs/figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-7-2.png" target="_blank">9008bd6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-f843a680a1fcd62a0dd1" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f843a680a1fcd62a0dd1">{"x":{"filter":"none","vertical":false,"caption":"<caption>disease_OMIM<\/caption>","data":[[],[],[],[],[]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>k<\/th>\n      <th>description<\/th>\n      <th>enrichmentRatio<\/th>\n      <th>userId<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"k","targets":1},{"name":"description","targets":2},{"name":"enrichmentRatio","targets":3},{"name":"userId","targets":4}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div class="datatables html-widget html-fill-item" id="htmlwidget-ba32e0541469b5ed4906" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ba32e0541469b5ed4906">{"x":{"filter":"none","vertical":false,"caption":"<caption>geneontology_Cellular_Component_noRedundant<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21"],["common","CD8_T","NK","Naive_B","CD4_T","common","Naive_B","Memory_B","Memory_B","CD8_T","Naive_B","NK","Memory_B","Naive_B","CD4_T","Memory_B","Memory_B","Memory_B","Memory_B","Memory_B","Memory_B"],["chaperone complex","chaperone complex","chaperone complex","chaperone complex","chaperone complex","myelin sheath","blood microparticle","immunological synapse","protein complex involved in cell adhesion","myelin sheath","myelin sheath","contractile fiber","COP9 signalosome","endocytic vesicle","extrinsic component of membrane","side of membrane","membrane region","receptor complex","extracellular matrix","vesicle lumen","cell-cell junction"],[68.40000000000001,46.5,32.8,27.6,13.8,11.9,9.4,8.699999999999999,7.9,7.4,6.2,5.5,5.4,4.3,4,3.3,3.1,2.9,2.5,2.2,2],["HSP90AB1;HSPA8;BAG3;STIP1","HSP90AB1;HSPA8;STIP1;PTGES3;BAG3;TCP1;CCT4","HSPA8;HSP90AB1;BAG3;STIP1;PTGES3;TCP1;CCT4","HSP90AB1;BAG3;HSPA8;STIP1;TCP1","HSP90AB1;HSPA8;BAG3;STIP1","HSP90AA1;HSPD1;HSPA8;UBC;STIP1","HSPA1A;HSPA6;HSPA8;HSPA1L;MSN","PRKCQ;GZMB;SKAP1;ZAP70;GZMA;CD3E;CD28;LGALS3;LAT;LCK;CD6;MYH9;HAVCR2","CD28;ITGA5;CTLA4;TSPAN32;ITGA1;ITGAX;ITGA6;ITGAM;ITGB1;ITGAL","HSP90AA1;HSPD1;HSPA8;UBC;STIP1;MSN;TCP1;ANXA2","HSP90AA1;HSPD1;HSPA8;STIP1;UBC;SOD1;TCP1;MSN","SYNE2;HSPB1;BAG3;SQSTM1;FLNA;MYOM2;AHNAK;TPM4;SVIL","HSPA1A;HSPA6;THEMIS;HSP90AB1;LAT;HSPA1L;MYH9;TESPA1","HSP90AA1;HSPH1;GNLY;UBC;SNAP23;ATP6V0A1;INPP5F;GIPC1;VIM;RAB20","FYN;SNX9;PIK3R1;SYTL2;ANXA2;SYNJ2;ATG2A;SMAD7;GNA15;CYTH3;NUCB1;ZC3H12A;ULK1;HIP1","IL7R;ANXA1;B2M;CD2;KLRD1;HSPA8;CD3D;RPS29;S100A6;TGFBR3;CD8A;CD3E;IL2RB;LDLR;ICOS;CD28;ITGA5;CD226;RPS26;YES1;CD5;CD3G;FCGR3A;FCER1G;NCAM1;LAG3;IL12RB2;CD40LG;CTLA4;LCK;TNFRSF4;THBS1;CD6;RGS1;RASA3;CD4;PDCD1;ITGA1;CXCR6;FCRL6;MATK;FYN;EFNA5;ADA;HLA-F;FASLG;TNFRSF9;CD244;NUCB1;FLT3LG;ITGA6;ITGAM;GNAI2;BCAP31;ITGB1","SKAP1;HSPD1;S100A10;ZAP70;CD8A;SERPINH1;CD226;ADGRG1;LAT;ATP2B4;STOM;LCP2;LCK;CBLB;SELPLG;TNFRSF1B;F2R;CD4;ITGA1;OPRM1;FYN;EFNA5;FASLG;LAX1;CTSD;KCNA3;ITGAM;GNAI2;FAS;PTCH1;ITGB1;PGK1;HMOX1;MAL","CD247;B2M;KLRD1;HSP90AB1;CD3D;SKAP1;HSPD1;TGFBR3;ZAP70;CD8A;CD3E;LDLR;ITGA5;KLRC1;KLRC2;CD8B;IL18RAP;CD3G;IL12RB2;PLXND1;TRAT1;CHRNA5;CD6;TNFRSF1B;TSPAN32;EPHA4;CD4;ITGA1;ITGAX;NOTCH1;PLXDC1;LIME1;ITGA6;ITGAM;ITGB1;ITGAL","ANXA1;S100A6;S100A10;TGFBR3;SERPINH1;SPON2;SPOCK2;LGALS3;LGALS1;NCAM1;THBS1;TIMP1;A2M;EFNA5;LRIG1;MXRA7;COL6A2;FGL2;ADAMTS10;IGFBP7;LINGO2;HAPLN3;SERPINE1;COLQ;VEGFA;CTSD;ITGA6;CTSC","HSPA6;GNLY;B2M;HSP90AA1;SRGN;HSPA8;HSP90AB1;HSPH1;CTSW;BIN2;S100A11;THBS1;TMSB4X;TIMP1;NEU1;A2M;ADA;FASLG;RAB27A;SERPINE1;VEGFA;CTSD;GMFG;FTL;CTSC;GPI","ANXA1;ITK;THEMIS;SKAP1;ZAP70;CD3E;ITGA5;JAML;LAT;FLNA;SMAD7;S100A11;LCP2;PARD6G;PTPRM;NECTIN3;AQP3;PERP;PLXDC1;MYH9;KIFC3;CFL1;CDHR3;ZYX;ITGA6;PIK3R1;VAMP5;ITGB1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>k<\/th>\n      <th>description<\/th>\n      <th>enrichmentRatio<\/th>\n      <th>userId<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"k","targets":1},{"name":"description","targets":2},{"name":"enrichmentRatio","targets":3},{"name":"userId","targets":4}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p><strong>Spleen Up-regulated genes</strong></p>
<p>Overall, spleen up-regulated genes are less enriched for GO terms at
a higher threshold at FDR &lt; 0.1.</p>
<p>The DE genes down-regulated in lung detected from memory B cells are
significantly enriched for asthma risk genes from KEGG pathway. The
overlapped genes are HLA genes and <em>CD40</em>. Their function in B
cells might be enhancing subsequent interaction with T cells.</p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-9-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9008bd6d61a5fd68cbc2639bec30281d654abacb/docs/figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-9-1.png" target="_blank">9008bd6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-9-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9008bd6d61a5fd68cbc2639bec30281d654abacb/docs/figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-9-2.png" target="_blank">9008bd6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-484ed42468d86de4a421" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-484ed42468d86de4a421">{"x":{"filter":"none","vertical":false,"caption":"<caption>disease_OMIM<\/caption>","data":[[],[],[],[],[]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>k<\/th>\n      <th>description<\/th>\n      <th>enrichmentRatio<\/th>\n      <th>userId<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"k","targets":1},{"name":"description","targets":2},{"name":"enrichmentRatio","targets":3},{"name":"userId","targets":4}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div class="datatables html-widget html-fill-item" id="htmlwidget-697d58abce2053a24549" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-697d58abce2053a24549">{"x":{"filter":"none","vertical":false,"caption":"<caption>geneontology_Cellular_Component_noRedundant<\/caption>","data":[["1","2","3","4"],["Memory_B","Memory_B","Memory_B","Memory_B"],["MHC protein complex","neuron spine","side of membrane","cell leading edge"],[6.6,2.1,1.8,1.7],["HLA-DQA1;CD74;HLA-DRA;HLA-DQB1;HLA-DMB;HLA-DRB5;HLA-DQA2;HLA-DOA","ATP2B1;RHOA;PPP3CA;PDE4B;PTEN;PTK2;GPM6A;ARHGAP32;APP;MAP1B;ZNF804A;CDK5R1;FCGR2B;ARHGAP44;DLG4;NEURL1;KCNN2;GPER1;KCNC3","HLA-DQA1;CD74;HLA-DRA;MS4A1;LYN;HLA-DQB1;TRAF3;SNX9;ATP2B1;RHOA;BLK;PPP3CA;PTEN;TNFRSF13C;ANTXR2;BORCS5;FER;FCER2;CD79A;CXCR5;CNR2;ALCAM;CLCN3;P2RX1;LRRK2;BTNL9;CD80;ABCA1;CD19;HLA-DRB5;KCNJ3;CD79B;ENOX2;CD24;CD40;INSR;HLA-DQA2;FCGR2B;IL5RA;SRC;DIABLO;DLG4;IGLL1;GNA11","SNX2;SNX9;CTNNB1;ATP2B1;RHOA;CLASP2;JMY;PTK2;PTPRK;CD2AP;FER;FGD2;PAK1;C2CD5;APP;CARMIL1;PSD3;KANK1;SH3RF1;SCIMP;CDC42BPB;EPB41L5;INSR;SH2B2;CDKL5;PDE9A;SRC;ARHGAP44;ARHGEF4;SORBS2;WWC1;MYO10;ENAH;WASF3;CTNNA3;GPER1;WASF1;KCNC3"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>k<\/th>\n      <th>description<\/th>\n      <th>enrichmentRatio<\/th>\n      <th>userId<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"k","targets":1},{"name":"description","targets":2},{"name":"enrichmentRatio","targets":3},{"name":"userId","targets":4}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="k-means-clustering-for-log2fc" class="section level3">
<h3>K-means clustering for log2FC</h3>
<p>We performed K-means clustering over log2FC for all genes with at
most one NA across cell types.</p>
<pre><code>[1] &quot;Number of genes in each k-mean cluster:&quot;</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
  35  847  158  634  718  469  292  816   35  160  683  865  921 1198  207  873 </code></pre>
<pre><code>[1] &quot;The pair-wise correlation of genes for most clusters form a distribution skewed to 1.&quot;</code></pre>
<p><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9008bd6d61a5fd68cbc2639bec30281d654abacb/docs/figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-12-1.png" target="_blank">9008bd6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Heatmap for average log2FC for each cluster</p>
<p>Clustering for effect sizes does not show cell type specificity
except for memory B cells.</p>
<p><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/9008bd6d61a5fd68cbc2639bec30281d654abacb/docs/figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-13-1.png" target="_blank">9008bd6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Heatmap for average log2FC for each cell-type (Memory B excluded)</p>
<p>Clustering for effect sizes shows stronger cell-type specificity.</p>
<p><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Several clusters were selected by having distinct cluster mean in one
cell type compared to the rest to perform GSEA. The pattern is less
clear to me. Currently all genes contained in each cluster were used to
perform GSEA, so I may try top hundreds of genes for the analysis.</p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-16-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/identify_lung_specific_transcriptomic_features.Rmd/unnamed-chunk-16-2.png" width="1152" style="display: block; margin: auto;" /></p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-29a56e939b13f92b7856" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-29a56e939b13f92b7856">{"x":{"filter":"none","vertical":false,"caption":"<caption>disease_OMIM<\/caption>","data":[[],[],[],[],[]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>k<\/th>\n      <th>description<\/th>\n      <th>enrichmentRatio<\/th>\n      <th>userId<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"k","targets":1},{"name":"description","targets":2},{"name":"enrichmentRatio","targets":3},{"name":"userId","targets":4}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div class="datatables html-widget html-fill-item" id="htmlwidget-457827ab1876f7c57926" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-457827ab1876f7c57926">{"x":{"filter":"none","vertical":false,"caption":"<caption>geneontology_Cellular_Component_noRedundant<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13"],["shared","shared","CD8T_hi7","NaiveB_hi","MemB_hi12","shared","MemB_hi3","MemB_hi12","CD8T_hi7","MemB_hi12","NaiveB_hi","MemB_hi12","NaiveB_hi"],["COP9 signalosome","blood microparticle","chaperone complex","immunological synapse","immunological synapse","ficolin-1-rich granule","extracellular matrix","side of membrane","myelin sheath","receptor complex","side of membrane","membrane region","membrane region"],[28,17,15.9,12.5,12.2,9.4,5.9,4.2,4.1,3.8,3.4,3.4,3.1],["HSPA1A;HSPA6;HSPA1L","HSPA1A;HSPA6;HSPA1L","HSP90AB1;HSPA8;STIP1;PTGES3;TCP1;CCT4","CD6;GZMA;CD28;LAT;CD3E;PRKCQ","GZMA;CD28;LAT;CD3E;PRKCQ","HSP90AA1;HSPA1A;HSPA6;LGALS3;FGL2","SERPINH1;THBS1;LGALS3;SERPINE1;FGL2","IL7R;ANXA1;IL2RB;CD5;KLRD1;IL12RB2;TNFRSF4;CD28;CD3D;CD2;CXCR6;MATK;FCER1G;CD3G;TNFRSF9;CD226;CD3E;ITGA1;YES1","HSPD1;MSN;HSPA8;UBC;STIP1;ANXA2;TCP1;ERMN;PITPNA;ATP6V1A;PRDX1","CD247;KLRD1;IL12RB2;IL18RAP;CD3D;PLXND1;CD3G;CD8B;CD3E;PLXDC1;TSPAN32;ITGA1;TRAT1","IL7R;ANXA1;CD5;CD6;KLRD1;CD8A;ADAM9;CD28;CD3D;ITGA5;CD2;CAPN2;CD3G;CD226;ERRFI1;CD3E;CD274;YES1","LCP2;F2R;S100A10;LAT;CD226;SELPLG;OPRM1;KCNA3;ATP2B4;ITGA1","TNFRSF1B;STOM;CD8A;LCP2;S100A10;CAPN2;CTSD;LAT;CD226;ARID3A;SELPLG"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>k<\/th>\n      <th>description<\/th>\n      <th>enrichmentRatio<\/th>\n      <th>userId<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"k","targets":1},{"name":"description","targets":2},{"name":"enrichmentRatio","targets":3},{"name":"userId","targets":4}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre><code>

`



&lt;br&gt;
&lt;p&gt;
&lt;button type=&quot;button&quot; class=&quot;btn btn-default btn-workflowr btn-workflowr-sessioninfo&quot;
  data-toggle=&quot;collapse&quot; data-target=&quot;#workflowr-sessioninfo&quot;
  style = &quot;display: block;&quot;&gt;
  &lt;span class=&quot;glyphicon glyphicon-wrench&quot; aria-hidden=&quot;true&quot;&gt;&lt;/span&gt;
  Session information
&lt;/button&gt;
&lt;/p&gt;

&lt;div id=&quot;workflowr-sessioninfo&quot; class=&quot;collapse&quot;&gt;
</code></pre>
<p>R version 4.2.0 (2022-04-22) Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)</p>
<p>Matrix products: default BLAS/LAPACK:
/software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so</p>
<p>locale: [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C LC_TIME=C<br />
[4] LC_COLLATE=C LC_MONETARY=C LC_MESSAGES=C<br />
[7] LC_PAPER=C LC_NAME=C LC_ADDRESS=C<br />
[10] LC_TELEPHONE=C LC_MEASUREMENT=C LC_IDENTIFICATION=C</p>
<p>attached base packages: [1] stats4 grid stats graphics grDevices
utils datasets [8] methods base</p>
<p>other attached packages: [1] ggVennDiagram_1.5.2
SingleCellExperiment_1.20.1 [3] ComplexHeatmap_2.14.0
htmltools_0.5.8.1<br />
[5] scales_1.3.0 colorRamp2_0.1.0<br />
[7] rtracklayer_1.58.0 tidyr_1.3.1<br />
[9] dplyr_1.1.4 rhdf5_2.42.1<br />
[11] SummarizedExperiment_1.28.0 Biobase_2.58.0<br />
[13] MatrixGenerics_1.10.0 Rcpp_1.0.14<br />
[15] Matrix_1.6-5 GenomicRanges_1.50.2<br />
[17] GenomeInfoDb_1.34.9 IRanges_2.32.0<br />
[19] S4Vectors_0.36.2 BiocGenerics_0.44.0<br />
[21] matrixStats_1.5.0 data.table_1.17.0<br />
[23] stringr_1.5.1 plyr_1.8.9<br />
[25] magrittr_2.0.3 ggplot2_3.5.2<br />
[27] gtable_0.3.6 gtools_3.9.5<br />
[29] gridExtra_2.3 ArchR_1.0.2</p>
loaded via a namespace (and not attached): [1] bitops_1.0-9 fs_1.6.5
doParallel_1.0.17<br />
[4] RColorBrewer_1.1-3 rprojroot_2.0.4 tools_4.2.0<br />
[7] bslib_0.9.0 DT_0.33 R6_2.6.1<br />
[10] colorspace_2.1-1 rhdf5filters_1.10.1 GetoptLong_1.0.5<br />
[13] withr_3.0.2 tidyselect_1.2.1 compiler_4.2.0<br />
[16] git2r_0.33.0 cli_3.6.4 Cairo_1.6-2<br />
[19] DelayedArray_0.24.0 labeling_0.4.3 sass_0.4.9<br />
[22] yulab.utils_0.2.0 digest_0.6.37 Rsamtools_2.14.0<br />
[25] rmarkdown_2.29 XVector_0.38.0 pkgconfig_2.0.3<br />
[28] fastmap_1.2.0 htmlwidgets_1.6.4 rlang_1.1.5<br />
[31] GlobalOptions_0.1.2 rstudioapi_0.17.1 gridGraphics_0.5-1<br />
[34] farver_2.1.2 shape_1.4.6 jquerylib_0.1.4<br />
[37] BiocIO_1.8.0 generics_0.1.3 jsonlite_2.0.0<br />
[40] crosstalk_1.2.1 BiocParallel_1.32.6 RCurl_1.98-1.17<br />
[43] ggplotify_0.1.2 GenomeInfoDbData_1.2.9 patchwork_1.3.0<br />
[46] munsell_0.5.1 Rhdf5lib_1.20.0 lifecycle_1.0.4<br />
[49] stringi_1.8.4 whisker_0.4.1 yaml_2.3.10<br />
[52] zlibbioc_1.44.0 parallel_4.2.0 promises_1.3.2<br />
[55] forcats_1.0.0 crayon_1.5.3 lattice_0.22-7<br />
[58] Biostrings_2.66.0 circlize_0.4.15 knitr_1.50<br />
[61] pillar_1.10.2 rjson_0.2.23 codetools_0.2-20<br />
[64] XML_3.99-0.18 glue_1.8.0 evaluate_1.0.3<br />
[67] ggfun_0.1.8 vctrs_0.6.5 png_0.1-8<br />
[70] httpuv_1.6.15 foreach_1.5.2 purrr_1.0.4<br />
[73] clue_0.3-66 cachem_1.1.0 xfun_0.52<br />
[76] restfulr_0.0.15 later_1.4.2 tibble_3.2.1<br />
[79] aplot_0.2.5 iterators_1.0.14 GenomicAlignments_1.34.1 [82]
cluster_2.1.8.1 workflowr_1.7.1<br />
```
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
