<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Single-cell multiomics analyses on lung immune cells</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Lung_scMultiomics_paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/Lung_scMultiomics_paper">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Single-cell multiomics analyses on lung
immune cells</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-11-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Lung_scMultiomics_paper/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxinhelabLungscMultiomicspapertree34eefdc2c063e46c0b71b8b2dd507d5a2a2d8f78targetblank34eefdca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/tree/34eefdc2c063e46c0b71b8b2dd507d5a2a2d8f78" target="_blank">34eefdc</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxinhelabLungscMultiomicspapertree34eefdc2c063e46c0b71b8b2dd507d5a2a2d8f78targetblank34eefdca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/tree/34eefdc2c063e46c0b71b8b2dd507d5a2a2d8f78" target="_blank">34eefdc</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  .RData
    Untracked:  ArchRLogs/
    Untracked:  AvgHiC_ABC_three_celltypes.RDS
    Untracked:  Figure1.pdf
    Untracked:  Lung_scMultiomics_paper.Rproj
    Untracked:  _workflowr.yml
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/ArchRLogs/
    Untracked:  analysis/Figure4.R
    Untracked:  analysis/Figure6.R
    Untracked:  analysis/Plots/
    Untracked:  analysis/Rplots.pdf
    Untracked:  analysis/TF_GRN_analysis.Rmd
    Untracked:  analysis/TF_GRN_analysis_cross_tissue.Rmd
    Untracked:  analysis/TF_GRN_analysis_old.Rmd
    Untracked:  analysis/about.knit.md
    Untracked:  analysis/archive.Rmd
    Untracked:  analysis/dictys/
    Untracked:  analysis/figures_for_grant_application.Rmd
    Untracked:  analysis/finalize_p2g_analysis.Rmd
    Untracked:  analysis/finalize_p2g_analysis_30CREs.Rmd
    Untracked:  analysis/finalize_p2g_analysis_50CREs.Rmd
    Untracked:  analysis/gene.Rmd
    Untracked:  analysis/gene_regulatory_network_analysis.Rmd
    Untracked:  analysis/glmpca.Rmd
    Untracked:  analysis/identify_lung_specific_epigenetic_features.Rmd
    Untracked:  analysis/link_peaks_to_genes.Rmd
    Untracked:  analysis/link_peaks_to_genes_filtered.Rmd
    Untracked:  analysis/link_peaks_to_genes_wilcoxon.Rmd
    Untracked:  analysis/make_figure_panels.Rmd
    Untracked:  analysis/make_figures.Rmd
    Untracked:  analysis/make_locus_plots.Rmd
    Untracked:  analysis/peak_to_gene_analyses.Rmd
    Untracked:  analysis/peak_to_gene_analyses_50CREs.Rmd
    Untracked:  analysis/preprocess_scRNA_seq_data.ipynb
    Untracked:  analysis/publication/
    Untracked:  analysis/run_dimension_reduction_glmPCA.ipynb
    Untracked:  analysis/run_peak2gene_linkage.Rmd
    Untracked:  analysis/summarize_candidate_genes.Rmd
    Untracked:  analysis/summarize_candidate_genes.nb.html
    Untracked:  code/compare_tau_star_calculations.R
    Untracked:  code/identify_asthma_CREs.R
    Untracked:  code/identify_marker_peaks_edgeR.R
    Untracked:  code/make_plots.R
    Untracked:  code/plot_heatmap_asthma_CREs.R
    Untracked:  code/preprocess/
    Untracked:  code/run_GO_enrichment.R
    Untracked:  code/run_GO_enrichment_GRN.R
    Untracked:  code/run_archR_p2g_analyses.R
    Untracked:  code/track_plots_mapgen.R
    Untracked:  code/utility_function.R
    Untracked:  code/utils_mapgen.R
    Untracked:  data/AvgHiC_ABC_B_ENCODE.RDS
    Untracked:  data/AvgHiC_ABC_CD4_T_Corces2016.RDS
    Untracked:  data/AvgHiC_ABC_CD4_T_Corces2016.txt
    Untracked:  data/AvgHiC_ABC_CD8_T_Corces2016.RDS
    Untracked:  data/SCENIC_plus_TF_gene.csv
    Untracked:  data/aoa_ePIPs.txt
    Untracked:  data/aoa_gene_scores.txt
    Untracked:  data/asthma_GWAS_snps_hg38.RDS
    Untracked:  data/asthma_combined_fine_mapping_gwas_L5.rds
    Untracked:  data/asthma_related_CREs_combined.RDS
    Untracked:  data/asthma_related_CREs_combined_all_p2g_links.RDS
    Untracked:  data/asthma_related_CREs_combined_all_p2g_long_tbl.RDS
    Untracked:  data/asthma_related_CREs_combined_all_p2g_long_tbl.tsv
    Untracked:  data/asthma_related_CREs_combined_heatmap.RDS
    Untracked:  data/asthma_related_CREs_combined_heatmap_order.RDS
    Untracked:  data/asthma_related_CREs_combined_p2g_full.RDS
    Untracked:  data/asthma_related_CREs_combined_p2g_max.RDS
    Untracked:  data/asthma_related_CREs_full.RDS
    Untracked:  data/asthma_related_CREs_heatmap.RDS
    Untracked:  data/asthma_related_CREs_high_ePIPs.RDS
    Untracked:  data/asthma_related_CREs_high_ePIPs_p2g_full.RDS
    Untracked:  data/asthma_related_CREs_p2g.RDS
    Untracked:  data/asthma_related_CREs_strong_links.RDS
    Untracked:  data/asthma_risk_genes_two_studies.txt
    Untracked:  data/coa_ePIPs.txt
    Untracked:  data/coa_gene_scores.txt
    Untracked:  data/color_codes.RDS
    Untracked:  data/comparing_min_pct_GO.RData
    Untracked:  data/lung_RNA_CPM_by_CellType.RDS
    Untracked:  data/p2g_res/
    Untracked:  data/p2g_v3/
    Untracked:  data/pseudobulk_rna_CPM.RDS
    Untracked:  data/sample_covariates.txt
    Untracked:  data/scE2G_gene_list.txt
    Untracked:  data/scE2G_signif_links.RDS
    Untracked:  data/u19_full_atac_cell_metadata.RDS
    Untracked:  data/u19_full_atac_cell_metadata_with_sampleID.RDS
    Untracked:  keep_keys.cfg
    Untracked:  output/BActivationTF_targets_L_MemB.txt
    Untracked:  output/CD4_T_spleen_bulk_DEG.txt
    Untracked:  output/CD8_T_spleen_bulk_DEG.txt
    Untracked:  output/CD9_T_spleen_bulk_DEG.txt
    Untracked:  output/CEBPD_targets_L_MemB.txt
    Untracked:  output/CXXC1_targets_CD4_T.txt
    Untracked:  output/Differential_accessibility_by_celltype.xlsx
    Untracked:  output/Differential_accessibility_cross_tissue.xlsx
    Untracked:  output/Differential_expression_cross_tissue.xlsx
    Untracked:  output/E2F4_targets_L_MemB.txt
    Untracked:  output/E2G_links_vs_ethan_scores.RDS
    Untracked:  output/E2G_target_genes_asthma_CREs.txt
    Untracked:  output/EGR2_ALL_L_MemB.txt
    Untracked:  output/EGR2_SOX5_RUNX1_L_MemB.txt
    Untracked:  output/EGR2_targets_L_MemB.txt
    Untracked:  output/ELK1_targets_L_MemB.txt
    Untracked:  output/ELK4_targets_CD4_T.txt
    Untracked:  output/Figure1.pdf
    Untracked:  output/Figure1_double.pdf
    Untracked:  output/Figure4_double.pdf
    Untracked:  output/GATA3_targets_L_MemB.txt
    Untracked:  output/Gene_set_enrichment_lung_upregulated_genes.xlsx
    Untracked:  output/Gene_set_enrichment_spleen_upregulated_genes.xlsx
    Untracked:  output/IRF8_targets_Th17.txt
    Untracked:  output/JUN_targets_Th17.txt
    Untracked:  output/MAF_targets_Th17.txt
    Untracked:  output/MAZ_targets_L_MemB.txt
    Untracked:  output/MEF2B_targets_L_MemB.txt
    Untracked:  output/MemB_cross_tissue_bulk_DEG.txt
    Untracked:  output/NaiveB_cross_tissue_bulk_DEG.txt
    Untracked:  output/Plot-UMAP-Discrepant_Genes-P2GLinks-GeneIntegrationScores.pdf
    Untracked:  output/Plot-UMAP-Discrepant_Genes-P2GLinks-GeneScores.pdf
    Untracked:  output/Plot-UMAP-E2G-GeneIntegration.pdf
    Untracked:  output/Plot-UMAP-E2G-GeneScores.pdf
    Untracked:  output/SP2_targets_CD4_T.txt
    Untracked:  output/TBP_targets_L_MemB.txt
    Untracked:  output/TF_postivie_regulators_L_MemB_DE.txt
    Untracked:  output/THAP1_targets_L_MemB.txt
    Untracked:  output/ZBTB6_ALL_L_MemB.txt
    Untracked:  output/all_targets_L_CD4_T_GRN.txt
    Untracked:  output/all_targets_L_GRN.txt
    Untracked:  output/all_targets_L_MemB.txt
    Untracked:  output/all_targets_L_Th17.txt
    Untracked:  output/all_targets_L_Th17_GRN.txt
    Untracked:  output/archR_all_genes_tested.RDS
    Untracked:  output/asthma_CRE_p2g_links_final_list.txt
    Untracked:  output/asthma_CREs_p2g_combined_tbl.txt
    Untracked:  output/asthma_CREs_p2g_combined_tbl_ALL.txt
    Untracked:  output/asthma_CREs_p2g_filtered_heatmap_inputs.RData
    Untracked:  output/asthma_E2G_linked_genes.txt
    Untracked:  output/asthma_E2G_linked_genes_full.txt
    Untracked:  output/asthma_candidate_risk_genes.xlsx
    Untracked:  output/asthma_candidate_risk_genes_full.xlsx
    Untracked:  output/co_activation_scE2G_multiome_tbl.txt
    Untracked:  output/co_activation_table_scE2G.txt
    Untracked:  output/co_activation_table_scE2G_gene_list.txt
    Untracked:  output/genomic_tracks.pdf
    Untracked:  output/genomic_tracks2.pdf
    Untracked:  output/genomic_tracks_CD4T_specific.pdf
    Untracked:  output/genomic_tracks_CXCR5_loci.pdf
    Untracked:  output/genomic_tracks_LRRC32_loci.pdf
    Untracked:  output/genomic_tracks_RPS25_loci.pdf
    Untracked:  output/genomic_tracks_not_shared_blood.pdf
    Untracked:  output/genomic_tracks_selected.pdf
    Untracked:  output/genomic_tracks_selected_loci.pdf
    Untracked:  output/genomic_tracks_selected_loci_80K.pdf
    Untracked:  output/genomic_tracks_selected_loci_scE2G.pdf
    Untracked:  output/lung_GRN_TFs.csv
    Untracked:  output/lung_GRN_regulatory_markers.csv
    Untracked:  output/marker_peaks_cross_tissue_DA.pdf
    Untracked:  output/marker_peaks_cross_tissue_DA_FOSB.pdf
    Untracked:  output/marker_peaks_cross_tissue_DA_HSPA1A.pdf
    Untracked:  output/master_regulators_GO_enrichment.RDS
    Untracked:  output/master_regulators_sig.RData
    Untracked:  output/p2g_57_links.txt
    Untracked:  output/p2g_links_ArchR.txt
    Untracked:  output/p2g_links_scE2G.txt
    Untracked:  output/p2g_links_scE2G_mapped_to_archR.txt
    Untracked:  output/p2g_links_scE2G_scATAC_mapped_to_archR.txt
    Untracked:  output/p2g_scE2G_scATAC_mapped_to_archR_peakset.txt
    Untracked:  output/scE2G_all_genes_tested.RDS
    Untracked:  output/scE2G_genes_vs_prior_genes.RDS
    Untracked:  output/scE2G_genome_wide_linked_genes.RDS
    Untracked:  output/smo_expression_markers_ImmuneLow.xlsx
    Untracked:  output/smo_expression_markers_LungAirWay.xlsx
    Untracked:  output/u19_multiomics
    Untracked:  plots/
    Untracked:  references.bib
    Untracked:  references_sim.bib
    Untracked:  simplify_bib.sh
    Untracked:  tables/

Unstaged changes:
    Modified:   README.md
    Modified:   analysis/heritability_enrichment_for_lung_open_chromatin.Rmd
    Modified:   analysis/identify_cell_types.Rmd
    Modified:   analysis/identify_lung_specific_transcriptomic_features.Rmd
    Modified:   analysis/test.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xinhe-lab/Lung_scMultiomics_paper/7933e286c12290c5819cce4621f8deec381b62de/docs/index.html" target="_blank">7933e28</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/abc6dc8573978dbb49bb594e2ea0b7ee5df73afc/analysis/index.Rmd" target="_blank">abc6dc8</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
<td>
TF regulation to explain cross-tissue DE genes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xinhe-lab/Lung_scMultiomics_paper/59c7c88a220c60c97e77485e1c861924a8620e20/docs/index.html" target="_blank">59c7c88</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xinhe-lab/Lung_scMultiomics_paper/007fdb91e953babc761c2e56bfeb33386a71b13b/docs/index.html" target="_blank">007fdb9</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/fa9188648ab80b79f1d697d6c4bd49dff5f2bb18/analysis/index.Rmd" target="_blank">fa91886</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-22
</td>
<td>
updated index page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xinhe-lab/Lung_scMultiomics_paper/497b42c341ba272e118cfec3270a9b21b27576f5/docs/index.html" target="_blank">497b42c</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xinhe-lab/Lung_scMultiomics_paper/9008bd6d61a5fd68cbc2639bec30281d654abacb/docs/index.html" target="_blank">9008bd6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/c40f0da66d1c349627657922366e98a1ce44c255/analysis/index.Rmd" target="_blank">c40f0da</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-14
</td>
<td>
DEG analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xinhe-lab/Lung_scMultiomics_paper/358af1236161da7ba2030a960a608fd1d2a8f65c/docs/index.html" target="_blank">358af12</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/53d9d851215d9f9bb57d92a6819b7fecbb3072ea/analysis/index.Rmd" target="_blank">53d9d85</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-13
</td>
<td>
updated index page
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/db6ddbbbea5fdbc8ae5525e528f88c6ecd51c2f2/analysis/index.Rmd" target="_blank">db6ddbb</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-12
</td>
<td>
started workflowr
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xinhe-lab/Lung_scMultiomics_paper/db6ddbbbea5fdbc8ae5525e528f88c6ecd51c2f2/docs/index.html" target="_blank">db6ddbb</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-12
</td>
<td>
started workflowr
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Previous study showed immune cells have a shared role in the
development of both AOA and COA based on heritability analysis on
ATAC-seq data of blood and lung cell types. Due to lack of resolution,
it is unclear if different immune subsets sampled from lung or other
tissues can have distinct roles in the genetics of asthma.</p>
</div>
<div id="hypotheses" class="section level2">
<h2>Hypotheses</h2>
<ul>
<li>Lung resident immune cells have distinct transcriptomic and
epigenetic features</li>
<li>Lung resident immune cells at a granular level show distinct roles
in the genetics of asthma.</li>
<li>Fine-mapped putative causal variants at asthma loci may regulate
enhancer activity through one or more lung immune subsets.</li>
</ul>
</div>
<div id="experiment-design" class="section level2">
<h2>Experiment design</h2>
<p>Single-cell multiomics on paired lung and spleen samples from 9
donors (6 controls and 3 cases).</p>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<p><a href="identify_cell_types.html">annotate cell types for
scMultiomics datasets</a></p>
<p><a
href="identify_lung_specific_transcriptomic_features.html">transcriptomic
features specific to lung</a></p>
<p><a
href="heritability_enrichment_for_lung_open_chromatin.html">heritability
enrichment for lung open chromatin regions</a></p>
<p><a href="link_peaks_to_genes.html">linking asthma-related CREs to
genes</a></p>
<p><a href="TF_mediated_cross_tissue_DE.html">Testing TFs mediated
cross-tissue GE differences</a></p>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
