<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-10-31" />

<title>testing TF mediating cross-tissue GE</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Lung_scMultiomics_paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/Lung_scMultiomics_paper">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">testing TF mediating cross-tissue GE</h1>
<h4 class="date">2025-10-31</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-11-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Lung_scMultiomics_paper/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250512code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250512)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250512code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250512)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxinhelabLungscMultiomicspapertree34eefdc2c063e46c0b71b8b2dd507d5a2a2d8f78targetblank34eefdca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/tree/34eefdc2c063e46c0b71b8b2dd507d5a2a2d8f78" target="_blank">34eefdc</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxinhelabLungscMultiomicspapertree34eefdc2c063e46c0b71b8b2dd507d5a2a2d8f78targetblank34eefdca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/tree/34eefdc2c063e46c0b71b8b2dd507d5a2a2d8f78" target="_blank">34eefdc</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  .RData
    Untracked:  ArchRLogs/
    Untracked:  AvgHiC_ABC_three_celltypes.RDS
    Untracked:  Figure1.pdf
    Untracked:  Lung_scMultiomics_paper.Rproj
    Untracked:  _workflowr.yml
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/ArchRLogs/
    Untracked:  analysis/Figure4.R
    Untracked:  analysis/Figure6.R
    Untracked:  analysis/Plots/
    Untracked:  analysis/Rplots.pdf
    Untracked:  analysis/TF_GRN_analysis.Rmd
    Untracked:  analysis/TF_GRN_analysis_cross_tissue.Rmd
    Untracked:  analysis/TF_GRN_analysis_old.Rmd
    Untracked:  analysis/about.knit.md
    Untracked:  analysis/archive.Rmd
    Untracked:  analysis/dictys/
    Untracked:  analysis/figures_for_grant_application.Rmd
    Untracked:  analysis/finalize_p2g_analysis.Rmd
    Untracked:  analysis/finalize_p2g_analysis_30CREs.Rmd
    Untracked:  analysis/finalize_p2g_analysis_50CREs.Rmd
    Untracked:  analysis/gene.Rmd
    Untracked:  analysis/gene_regulatory_network_analysis.Rmd
    Untracked:  analysis/glmpca.Rmd
    Untracked:  analysis/identify_lung_specific_epigenetic_features.Rmd
    Untracked:  analysis/link_peaks_to_genes.Rmd
    Untracked:  analysis/link_peaks_to_genes_filtered.Rmd
    Untracked:  analysis/link_peaks_to_genes_wilcoxon.Rmd
    Untracked:  analysis/make_figure_panels.Rmd
    Untracked:  analysis/make_figures.Rmd
    Untracked:  analysis/make_locus_plots.Rmd
    Untracked:  analysis/peak_to_gene_analyses.Rmd
    Untracked:  analysis/peak_to_gene_analyses_50CREs.Rmd
    Untracked:  analysis/preprocess_scRNA_seq_data.ipynb
    Untracked:  analysis/publication/
    Untracked:  analysis/run_dimension_reduction_glmPCA.ipynb
    Untracked:  analysis/run_peak2gene_linkage.Rmd
    Untracked:  analysis/summarize_candidate_genes.Rmd
    Untracked:  analysis/summarize_candidate_genes.nb.html
    Untracked:  code/compare_tau_star_calculations.R
    Untracked:  code/identify_asthma_CREs.R
    Untracked:  code/identify_marker_peaks_edgeR.R
    Untracked:  code/make_plots.R
    Untracked:  code/plot_heatmap_asthma_CREs.R
    Untracked:  code/preprocess/
    Untracked:  code/run_GO_enrichment.R
    Untracked:  code/run_GO_enrichment_GRN.R
    Untracked:  code/run_archR_p2g_analyses.R
    Untracked:  code/track_plots_mapgen.R
    Untracked:  code/utility_function.R
    Untracked:  code/utils_mapgen.R
    Untracked:  data/AvgHiC_ABC_B_ENCODE.RDS
    Untracked:  data/AvgHiC_ABC_CD4_T_Corces2016.RDS
    Untracked:  data/AvgHiC_ABC_CD4_T_Corces2016.txt
    Untracked:  data/AvgHiC_ABC_CD8_T_Corces2016.RDS
    Untracked:  data/SCENIC_plus_TF_gene.csv
    Untracked:  data/aoa_ePIPs.txt
    Untracked:  data/aoa_gene_scores.txt
    Untracked:  data/asthma_GWAS_snps_hg38.RDS
    Untracked:  data/asthma_combined_fine_mapping_gwas_L5.rds
    Untracked:  data/asthma_related_CREs_combined.RDS
    Untracked:  data/asthma_related_CREs_combined_all_p2g_links.RDS
    Untracked:  data/asthma_related_CREs_combined_all_p2g_long_tbl.RDS
    Untracked:  data/asthma_related_CREs_combined_all_p2g_long_tbl.tsv
    Untracked:  data/asthma_related_CREs_combined_heatmap.RDS
    Untracked:  data/asthma_related_CREs_combined_heatmap_order.RDS
    Untracked:  data/asthma_related_CREs_combined_p2g_full.RDS
    Untracked:  data/asthma_related_CREs_combined_p2g_max.RDS
    Untracked:  data/asthma_related_CREs_full.RDS
    Untracked:  data/asthma_related_CREs_heatmap.RDS
    Untracked:  data/asthma_related_CREs_high_ePIPs.RDS
    Untracked:  data/asthma_related_CREs_high_ePIPs_p2g_full.RDS
    Untracked:  data/asthma_related_CREs_p2g.RDS
    Untracked:  data/asthma_related_CREs_strong_links.RDS
    Untracked:  data/asthma_risk_genes_two_studies.txt
    Untracked:  data/coa_ePIPs.txt
    Untracked:  data/coa_gene_scores.txt
    Untracked:  data/color_codes.RDS
    Untracked:  data/comparing_min_pct_GO.RData
    Untracked:  data/lung_RNA_CPM_by_CellType.RDS
    Untracked:  data/p2g_res/
    Untracked:  data/p2g_v3/
    Untracked:  data/pseudobulk_rna_CPM.RDS
    Untracked:  data/sample_covariates.txt
    Untracked:  data/scE2G_gene_list.txt
    Untracked:  data/scE2G_signif_links.RDS
    Untracked:  data/u19_full_atac_cell_metadata.RDS
    Untracked:  data/u19_full_atac_cell_metadata_with_sampleID.RDS
    Untracked:  keep_keys.cfg
    Untracked:  output/BActivationTF_targets_L_MemB.txt
    Untracked:  output/CD4_T_spleen_bulk_DEG.txt
    Untracked:  output/CD8_T_spleen_bulk_DEG.txt
    Untracked:  output/CD9_T_spleen_bulk_DEG.txt
    Untracked:  output/CEBPD_targets_L_MemB.txt
    Untracked:  output/CXXC1_targets_CD4_T.txt
    Untracked:  output/Differential_accessibility_by_celltype.xlsx
    Untracked:  output/Differential_accessibility_cross_tissue.xlsx
    Untracked:  output/Differential_expression_cross_tissue.xlsx
    Untracked:  output/E2F4_targets_L_MemB.txt
    Untracked:  output/E2G_links_vs_ethan_scores.RDS
    Untracked:  output/E2G_target_genes_asthma_CREs.txt
    Untracked:  output/EGR2_ALL_L_MemB.txt
    Untracked:  output/EGR2_SOX5_RUNX1_L_MemB.txt
    Untracked:  output/EGR2_targets_L_MemB.txt
    Untracked:  output/ELK1_targets_L_MemB.txt
    Untracked:  output/ELK4_targets_CD4_T.txt
    Untracked:  output/Figure1.pdf
    Untracked:  output/Figure1_double.pdf
    Untracked:  output/Figure4_double.pdf
    Untracked:  output/GATA3_targets_L_MemB.txt
    Untracked:  output/Gene_set_enrichment_lung_upregulated_genes.xlsx
    Untracked:  output/Gene_set_enrichment_spleen_upregulated_genes.xlsx
    Untracked:  output/IRF8_targets_Th17.txt
    Untracked:  output/JUN_targets_Th17.txt
    Untracked:  output/MAF_targets_Th17.txt
    Untracked:  output/MAZ_targets_L_MemB.txt
    Untracked:  output/MEF2B_targets_L_MemB.txt
    Untracked:  output/MemB_cross_tissue_bulk_DEG.txt
    Untracked:  output/NaiveB_cross_tissue_bulk_DEG.txt
    Untracked:  output/Plot-UMAP-Discrepant_Genes-P2GLinks-GeneIntegrationScores.pdf
    Untracked:  output/Plot-UMAP-Discrepant_Genes-P2GLinks-GeneScores.pdf
    Untracked:  output/Plot-UMAP-E2G-GeneIntegration.pdf
    Untracked:  output/Plot-UMAP-E2G-GeneScores.pdf
    Untracked:  output/SP2_targets_CD4_T.txt
    Untracked:  output/TBP_targets_L_MemB.txt
    Untracked:  output/TF_postivie_regulators_L_MemB_DE.txt
    Untracked:  output/THAP1_targets_L_MemB.txt
    Untracked:  output/ZBTB6_ALL_L_MemB.txt
    Untracked:  output/all_targets_L_CD4_T_GRN.txt
    Untracked:  output/all_targets_L_GRN.txt
    Untracked:  output/all_targets_L_MemB.txt
    Untracked:  output/all_targets_L_Th17.txt
    Untracked:  output/all_targets_L_Th17_GRN.txt
    Untracked:  output/archR_all_genes_tested.RDS
    Untracked:  output/asthma_CRE_p2g_links_final_list.txt
    Untracked:  output/asthma_CREs_p2g_combined_tbl.txt
    Untracked:  output/asthma_CREs_p2g_combined_tbl_ALL.txt
    Untracked:  output/asthma_CREs_p2g_filtered_heatmap_inputs.RData
    Untracked:  output/asthma_E2G_linked_genes.txt
    Untracked:  output/asthma_E2G_linked_genes_full.txt
    Untracked:  output/asthma_candidate_risk_genes.xlsx
    Untracked:  output/asthma_candidate_risk_genes_full.xlsx
    Untracked:  output/co_activation_scE2G_multiome_tbl.txt
    Untracked:  output/co_activation_table_scE2G.txt
    Untracked:  output/co_activation_table_scE2G_gene_list.txt
    Untracked:  output/genomic_tracks.pdf
    Untracked:  output/genomic_tracks2.pdf
    Untracked:  output/genomic_tracks_CD4T_specific.pdf
    Untracked:  output/genomic_tracks_CXCR5_loci.pdf
    Untracked:  output/genomic_tracks_LRRC32_loci.pdf
    Untracked:  output/genomic_tracks_RPS25_loci.pdf
    Untracked:  output/genomic_tracks_not_shared_blood.pdf
    Untracked:  output/genomic_tracks_selected.pdf
    Untracked:  output/genomic_tracks_selected_loci.pdf
    Untracked:  output/genomic_tracks_selected_loci_80K.pdf
    Untracked:  output/genomic_tracks_selected_loci_scE2G.pdf
    Untracked:  output/lung_GRN_TFs.csv
    Untracked:  output/lung_GRN_regulatory_markers.csv
    Untracked:  output/marker_peaks_cross_tissue_DA.pdf
    Untracked:  output/marker_peaks_cross_tissue_DA_FOSB.pdf
    Untracked:  output/marker_peaks_cross_tissue_DA_HSPA1A.pdf
    Untracked:  output/master_regulators_GO_enrichment.RDS
    Untracked:  output/master_regulators_sig.RData
    Untracked:  output/p2g_57_links.txt
    Untracked:  output/p2g_links_ArchR.txt
    Untracked:  output/p2g_links_scE2G.txt
    Untracked:  output/p2g_links_scE2G_mapped_to_archR.txt
    Untracked:  output/p2g_links_scE2G_scATAC_mapped_to_archR.txt
    Untracked:  output/p2g_scE2G_scATAC_mapped_to_archR_peakset.txt
    Untracked:  output/scE2G_all_genes_tested.RDS
    Untracked:  output/scE2G_genes_vs_prior_genes.RDS
    Untracked:  output/scE2G_genome_wide_linked_genes.RDS
    Untracked:  output/smo_expression_markers_ImmuneLow.xlsx
    Untracked:  output/smo_expression_markers_LungAirWay.xlsx
    Untracked:  output/u19_multiomics
    Untracked:  plots/
    Untracked:  references.bib
    Untracked:  references_sim.bib
    Untracked:  simplify_bib.sh
    Untracked:  tables/

Unstaged changes:
    Modified:   README.md
    Modified:   analysis/heritability_enrichment_for_lung_open_chromatin.Rmd
    Modified:   analysis/identify_cell_types.Rmd
    Modified:   analysis/identify_lung_specific_transcriptomic_features.Rmd
    Modified:   analysis/test.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/TF_mediated_cross_tissue_DE.Rmd</code>) and HTML
(<code>docs/TF_mediated_cross_tissue_DE.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/34eefdc2c063e46c0b71b8b2dd507d5a2a2d8f78/analysis/TF_mediated_cross_tissue_DE.Rmd" target="_blank">34eefdc</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
<td>
TF regulation to explain cross-tissue DE genes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xinhe-lab/Lung_scMultiomics_paper/7933e286c12290c5819cce4621f8deec381b62de/docs/TF_mediated_cross_tissue_DE.html" target="_blank">7933e28</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/abc6dc8573978dbb49bb594e2ea0b7ee5df73afc/analysis/TF_mediated_cross_tissue_DE.Rmd" target="_blank">abc6dc8</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
<td>
TF regulation to explain cross-tissue DE genes
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="cross-tissue-de-tested-in-bulk-by-deseq2"
class="section level2">
<h2>Cross-tissue DE tested in bulk by DESeq2</h2>
<p>100-650 cross-tissue DE genes were found across cell types with
single-cell approach. About half amount were detected with bulk
approach, except for Memory B cells.</p>
<pre><code>          CD4-T CD8-T NK Memory-B Naive-B
lung_up      26    17 27      273      24
spleen_up    41    46 38      281       1</code></pre>
<p>Top 10 lung-upregulated genes in each cell type</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-8d99af577d77a3f6bd29" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-8d99af577d77a3f6bd29">{"x":{"filter":"none","vertical":false,"caption":"<caption>Lung up-regulated DE genes<\/caption>","data":[["HSPA6","BAG3","CACYBP","DOK2","MSN","STIP1","PLIN2","PFKFB3","CCL5","MOB4"],["SERPINH1","DOK2","MSN","UGP2","IRF4","MOB4","PLIN2","HSPA4","ABHD3","ZFYVE28"],["HSPA6","BAG3","GNLY","CACYBP","HSPA1B","GBP2","HSPE1","STIP1","HSP90AA1","DOK2"],["GNLY","ID2","CD247","CCL5","CCL4","PRKCH","NIBAN1","NKG7","ITK","RORA"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>CD4-T<\/th>\n      <th>CD8-T<\/th>\n      <th>NK<\/th>\n      <th>Memory-B<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"name":"CD4-T","targets":0},{"name":"CD8-T","targets":1},{"name":"NK","targets":2},{"name":"Memory-B","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="datatables html-widget html-fill-item" id="htmlwidget-206ea766c87da472b6da" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-206ea766c87da472b6da">{"x":{"filter":"none","vertical":false,"caption":"<caption>Spleen up-regulated DE genes<\/caption>","data":[["BANK1","EBF1","ARHGAP24","CDK14","AFF3","HLA-DRA","COBLL1","CD22","PAX5","CCSER1"],["BANK1","AFF3","ARHGAP24","GALNTL6","MS4A1","MEF2C","PAX5","EGR1","CD22","CD79A"],["AFF3","BANK1","PTPRD","ME1","HIPK1-AS1","EBF1","ADAMTS17","FOSB","NAALADL2","WDFY4"],["CR2","CR1","AC098829.1","RGS7","CCSER1","PTPRK","KCNJ3","PRKG1","MAST4","STEAP1B"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>CD4-T<\/th>\n      <th>CD8-T<\/th>\n      <th>NK<\/th>\n      <th>Memory-B<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"name":"CD4-T","targets":0},{"name":"CD8-T","targets":1},{"name":"NK","targets":2},{"name":"Memory-B","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div
id="lung-upregulated-genes-were-enriched-for-go-terms-with-immune-related-functions"
class="section level2">
<h2>Lung-upregulated genes were enriched for GO terms with
immune-related functions</h2>
<p>Lung-upregulated genes in memory B cells and CD4+ T cells share many
pathways related to cell activation and responses to cytokines. These
results were similar as previous ones with single-cell approach; however
the signals were stronger with bulk approach.</p>
<p><img src="figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/7933e286c12290c5819cce4621f8deec381b62de/docs/figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-4-1.png" target="_blank">7933e28</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="spleen-upregulated-genes-were-strictly-enriched-in-b-cell-functions"
class="section level2">
<h2>Spleen-upregulated genes were strictly enriched in B cell
functions</h2>
<p>Somehow spleen-upregulated genes in T cells are enriched for B cell
activation.</p>
<p><img src="figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/7933e286c12290c5819cce4621f8deec381b62de/docs/figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-5-1.png" target="_blank">7933e28</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In summary, genes with higher expression in lungs are broadly
enriched for T cell related functions, while those in spleens are
broadly enriched for B cell related functions. We still see lung
up-regulated genes enriched in pathways of heat-shock protein genes.
Overall, the GO enrichment results are consistent between bulk or
single-cell approach.</p>
</div>
<div id="testing-whether-tfs-mediate-cross-tissue-ge-differences"
class="section level2">
<h2>Testing whether TFs mediate cross-tissue GE differences</h2>
<p>We have shown that chromatin accessibility for cross-tissue DE genes
are very similar. Thus, some other mechanisms may drive the differences,
which motivates us to test whether TF activity is responsible for the
up-regulation of genes in lung.</p>
<p>Cell composition for both tissues <img
src="assets/rna_cell_composition.png" alt="cell comosition" /></p>
<p>We examined GRNs for B cells because both tissues have relatively
similar number of cells. Though half amount of memory B cells found in
lung compared to spleen, we still detected several lung TFs with higher
activity relative to spleen. The differences in naive B cells are small,
which is consistent with lack of differences in cross-tissue DE genes in
naive B cells.</p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-7-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/7933e286c12290c5819cce4621f8deec381b62de/docs/figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-7-1.png" target="_blank">7933e28</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-7-2.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/7933e286c12290c5819cce4621f8deec381b62de/docs/figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-7-2.png" target="_blank">7933e28</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Let’s focus on TFs specific to lung memory B cells and ask whether
they explain up-regulated genes in lungs</p>
<div id="testing-whether-lung-specific-tfs-regulated-more-de-genes"
class="section level3">
<h3>Testing whether lung-specific TFs regulated more DE genes</h3>
<p>Here I performed the tests on DE genes identified using single-cell
or bulk approach. I identified lung-specific TFs by only requiring its
expression to be 1.5 fold higher in lung than spleen. I relaxed the
differential regulation criteria because the quantity of target counts
do not truly reflect the difference.</p>
<pre><code>[1] &quot;Results for single-cell approach DE genes:&quot;</code></pre>
<pre><code>      p_value odds_ratio conf_low conf_high        FDR
CEBPD 1.3e-03       1.69     1.23      2.29 0.02643333
IRF1    8e-04       1.79     1.26      2.50 0.02643333
RUNX2 1.2e-03       1.88     1.27      2.72 0.02643333</code></pre>
<pre><code>[1] &quot;Results for bulk approach DE genes:&quot;</code></pre>
<pre><code>      p_value odds_ratio conf_low conf_high        FDR
GATA3   2e-02       3.41     1.07      8.41 0.24400000
PRDM1 3.4e-03       3.48     1.45      7.23 0.06913333</code></pre>
</div>
<div
id="examining-the-effect-sizes-between-tf-regulation-and-differential-expression"
class="section level3">
<h3>Examining the effect sizes between TF regulation and differential
expression</h3>
<p>As we identified lung-specific TFs enriched for DE genes, we further
asked if TFs that have stronger links tend to have DE genes with larger
differences between lung and spleen. Let’s use gene expression in spleen
as baseline. If specific lung TFs drive gene expression in lung to be
higher than baseline, we would expect to higher TF activity leads to
higher expression.</p>
<p><strong>Method</strong> For each lung-specific TF, we identified lung
up-regulated genes in its network. Then we plotted TF strength and
differential expression and showed the fitted line.</p>
<p>Bulk DE genes</p>
<p><img src="figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/7933e286c12290c5819cce4621f8deec381b62de/docs/figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-10-1.png" target="_blank">7933e28</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-2">
Past versions of unnamed-chunk-10-2.png
</button>
</p>
<div id="fig-unnamed-chunk-10-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/7933e286c12290c5819cce4621f8deec381b62de/docs/figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-10-2.png" target="_blank">7933e28</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[[1]]
NULL

[[2]]
NULL</code></pre>
<p>Single-cell DE genes</p>
<p><img src="figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/7933e286c12290c5819cce4621f8deec381b62de/docs/figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-11-1.png" target="_blank">7933e28</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-2">
Past versions of unnamed-chunk-11-2.png
</button>
</p>
<div id="fig-unnamed-chunk-11-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/7933e286c12290c5819cce4621f8deec381b62de/docs/figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-11-2.png" target="_blank">7933e28</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-3">
Past versions of unnamed-chunk-11-3.png
</button>
</p>
<div id="fig-unnamed-chunk-11-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xinhe-lab/Lung_scMultiomics_paper/blob/7933e286c12290c5819cce4621f8deec381b62de/docs/figure/TF_mediated_cross_tissue_DE.Rmd/unnamed-chunk-11-3.png" target="_blank">7933e28</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-11-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL</code></pre>
<p>We saw positive trends for the two TFs enriched for bulk-level DE
genes, but much weaker trend for the three TFs enriched for single-cell
level DE genes.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggrepel_0.9.6     scales_1.4.0      data.table_1.17.4 ggplot2_3.5.2    
[5] tidyr_1.3.1       dplyr_1.1.4      

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.14        RColorBrewer_1.1-3 pillar_1.10.2      bslib_0.9.0       
 [5] compiler_4.2.0     later_1.4.2        jquerylib_0.1.4    git2r_0.33.0      
 [9] workflowr_1.7.1    tools_4.2.0        digest_0.6.37      gtable_0.3.6      
[13] jsonlite_2.0.0     evaluate_1.0.3     lifecycle_1.0.4    tibble_3.2.1      
[17] pkgconfig_2.0.3    rlang_1.1.6        cli_3.6.5          rstudioapi_0.17.1 
[21] crosstalk_1.2.1    yaml_2.3.10        xfun_0.52          fastmap_1.2.0     
[25] withr_3.0.2        stringr_1.5.1      knitr_1.50         htmlwidgets_1.6.4 
[29] generics_0.1.4     fs_1.6.6           vctrs_0.6.5        sass_0.4.10       
[33] DT_0.33            grid_4.2.0         rprojroot_2.0.4    tidyselect_1.2.1  
[37] glue_1.8.0         R6_2.6.1           rmarkdown_2.29     farver_2.1.2      
[41] purrr_1.0.4        magrittr_2.0.3     whisker_0.4.1      promises_1.3.2    
[45] htmltools_0.5.8.1  dichromat_2.0-0.1  httpuv_1.6.16      labeling_0.4.3    
[49] stringi_1.8.4      cachem_1.1.0      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
