---
title: "Annotating cell types for sc-multiomics dataset"
output:
  html_document: default
  word_document: default
  pdf_document: default
date: '`r Sys.Date()`'
author: 'Jing Gu'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE)
library(dplyr)
library(ArchR)
library(liftOver)
library(eulerr)
library(ggrepel)
library(cowplot)
work.dir<-"~/cluster/projects/u19_multiomics"
set.seed(10)
options(repr.plot.width=16, repr.plot.height=10)
proj <- loadArchRProject("output/u19_multiomics/analyses/u19_peaks_tissue/")
ct_order <-
  c(
  "CD8/CD4_T",
  "Naive_B",
  "Memory_B",
  "NK",
  "CD8_T",
  "CD4_T",
  "Th17/CD4_T",
  "Treg",
  "Other"
)

```

# Identifying lung and spleen immune cell types


## Annotate clusters in RNA-seq data with marker gene expression

UMAP for gene expression of all the cells, split by tissue of origin, with clusters annotated by CellTypist, a machine learning tool developed to predict cell types based on the expression of marker genes. 

![](assets/umap_scRNA.png)

Dotplot for expression levels of CD69 and marker genes for immune subsets

![](assets/dotplot_GE_markers.png)

### Cell composition across samples

```{r}
seRNA <- readRDS("~/cluster/projects/u19_multiomics/analyses/u19_seRNA_summarized_experiment.RDS") 
colData(seRNA)$CellType <- 
factor(seRNA$majority_voting,
       labels = c("NK",
                  "NK", "Other", 
                  "Other", "Other",
                  "Memory_B","Naive_B", "Other",
                  "Treg", "CD4_T", 
                  "CD8_T", "Th17")
       )
colData(seRNA)$CellType <- factor(
    colData(seRNA)$CellType,
    levels = 
            c("Other", "Treg", "Th17", "CD4_T",
              "CD8_T", "NK", "Memory_B", "Naive_B"))
pal <- paletteDiscrete(values = colData(seRNA)$CellType)
```

```{r, fig.width = 10}
sample_cov <- fread("~/cluster/projects/u19_multiomics/output/sample_covariates.txt")
rnameta<-readRDS("~/cluster/projects/u19_multiomics/fastTopics/u19_cell_metadata.RDS")
rnameta$major <- factor(rnameta$major, levels = rev(names(pal)))
rnameta$core_ID <- factor(rnameta$full.ident, 
                          labels = c(paste0("SMO", "-", c(2,3,5,7,8,9)),
                                     paste0("SMO", "-", c(1, 4, 10))))
rnameta <- left_join(rnameta, sample_cov[, c("ROBI", "core_ID", "cells")])
rnameta$SampleID <- paste0(rnameta$cells, "_", 
                           factor(rnameta$ROBI, 
                                   labels = c(1, 4, 3, 5, 6, 2)))

#rnameta<-readRDS("output/u19_multiomics/output/u19_matched_rna_atac_cell_metadata.RDS")
#rnameta$Sample <- factor(rnameta$Sample, labels = c("Lung", "Spleen"))

ggplot(rnameta,
       aes(y = paste0(SampleID),
           fill = major)) + 
  geom_bar(position = "fill") + 
  scale_fill_manual(values = pal) + 
  xlab("Cell Proportion") + ylab("") + 
  theme_ArchR(legendPosition = "right", legendTextSize = 15, baseSize = 15) +
  ggtitle("Spleen and lung have large differences in cell composition")
```

## Annotate clusters of ATAC-seq data with majority voting

### Sample QC 

COB-5 sample has a small cluster of cells with low number of fragments, so top 5K cells ranked by number of fragments were retained. As a result, we see more consistent distribution from the rigid plots of TSS enrichment and log10 of number of fragments across samples. 
```{r message = F}
p1 <- plotGroups(
    ArchRProj = proj, 
    groupBy = "Sample", 
    colorBy = "cellColData", 
    name = "TSSEnrichment",
  plotAs = "ridges",
    baseSize = 12
)

p2 <- plotGroups(
    ArchRProj = proj, 
    groupBy = "Sample", 
    colorBy = "cellColData", 
  name = "log10(nFrags)",
  plotAs = "ridges",
    baseSize = 12
)

ggAlignPlots(p1, p2, type = "h") 
```

Ridge plots for TSS enrichment and log10(nFrags) across samples show a relatively consistent distribution.
```{r eval = F}
#too slow
p3 <- plotFragmentSizes(ArchRProj = proj)
p4 <- plotTSSEnrichment(ArchRProj = proj)
ggAlignPlots(p3, p4, type = "h")
```

Fragment size distributions are variable across samples, but overall enriched for sizes of one or two nucleosomes

![](assets/atac_seq_fragment_size.png)

Clustering of ATAC-seq data shows no strong bias to one batch or one sample.

```{r message = F, fig.width = 8}
proj@cellColData$Batch <- ifelse(grepl("SMO", proj@cellColData$Sample), "1", "0")
p1<-plotEmbedding(ArchRProj = proj, colorBy = "cellColData", name = "Batch", embedding = "UMAP") + theme_ArchR(legendTextSize = 12)
p2<-plotEmbedding(ArchRProj = proj, colorBy = "cellColData", name = "Tissue", embedding = "UMAP") + 
  theme_ArchR(legendTextSize = 12)
p3<-plotEmbedding(ArchRProj = proj, colorBy = "cellColData", name = "Status", embedding = "UMAP") + 
  theme_ArchR(legendTextSize = 12)
p4<-plotEmbedding(ArchRProj = proj, colorBy = "cellColData", name = "Sample", embedding = "UMAP", ) + 
  theme_ArchR(legendTextSize = 8)
ggAlignPlots(p1, p4, type = "h")
```

Clustering of ATAC-seq data also shows no distinct clusters for either tissue or disease status.

```{r, message = F, fig.width = 8}
ggAlignPlots(p2, p3, type = "h")
```

### Cell composition for each cluster of ATAC-seq data 

Majority of clusters were dominated by one single cell type from matched RNA-seq, while two clusters show some ambiguity and so labeled as "CD8/CD4_T" and "Th17/CD4_T".

```{r, message = F, fig.width = 9}
proj@cellColData$PredCellType <-
  factor(proj@cellColData$PredCellType,
         levels = rev(ct_order))

plotEmbedding(ArchRProj = proj, colorBy = "cellColData", 
              name = "Clusters", embedding = "UMAP") + 
  theme(legend.position = "none")

pal_atac <- paletteDiscrete(values = proj@cellColData$PredCellType)

ggplot(proj@cellColData,
       aes(x = Clusters,
           fill = CellType)) + 
  geom_bar() + 
  scale_fill_manual(values = pal) + 
  theme_classic(base_size = 12) + 
  ggtitle("Majority of clusters are dominated by a single cell type defined by matched RNA-seq")
```

UMAP for ATAC-seq data from 100K cells

* left plot - cell labels before majority-voting

* right plot - cell labels after majority-voting
```{r, message = F, fig.width = 8}
pal <- paletteDiscrete(values = colData(seRNA)$CellType)
p1 <- plotEmbedding(ArchRProj = proj, colorBy = "cellColData", 
              name = "CellType", embedding = "UMAP",
              pal=pal)
p2 <-plotEmbedding(ArchRProj = proj, colorBy = "cellColData", 
              name = "PredCellType", embedding = "UMAP",
              pal = pal_atac)
ggAlignPlots(p1, p2, type = "h")
```

### Cell composition across samples

```{r, eval = F}
metadata <- proj@cellColData
metadata$core_ID <- metadata$Sample
metadata <- left_join(data.frame(metadata), sample_cov[, c("ROBI", "core_ID")])
metadata$SampleID <- paste0(metadata$Tissue, "_", 
                            factor(metadata$ROBI, 
                                   labels = c(1, 4, 3, 5, 6, 7, 2, 8, 9)))
metadata$PredCellType <- factor(metadata$PredCellType,
                                levels = rev(names(pal_atac)))
metadata$predictedGroup_Un <- factor(
  metadata$predictedGroup_Un,
  labels = c("NK", "NK", "Other", "Other",
             "Other", "Memory_B", "Naive_B",
             "Treg", "CD4_T", "CD8_T", "Th17") 
)
metadata$predictedGroup_Un <- factor(metadata$predictedGroup_Un,
                                levels = rev(names(pal)))
```

```{r, fig.height = 8, fig.width = 8}
metadata <- readRDS("~/projects/Lung_scMultiomics_paper/data/u19_full_atac_cell_metadata.RDS")
ggplot(metadata %>% dplyr::filter(SampleID != "Spleen_8") %>%
         mutate(SampleID = factor(SampleID,
                                  levels = c(paste0("Lung_", c(1:4,6,8,5,7,9)),
                                             paste0("Spleen_", c(1:3,6,7,9))),
                                  labels = c(paste0("L", c(1:4,6,8), "(C)"),
                                             paste0("L",c(5,7,9), "(A)"),
                                              paste0("S", c(1:3,6), "(C)"),
                                             paste0("S", c(7,9), "(A)")))),
       aes(x = SampleID,
           fill = PredCellType)) + 
  geom_bar(position = "fill") + 
  scale_fill_manual(values = pal_atac) + 
  ylab("Cell Proportion") + xlab("") + 
  guides(fill = guide_legend(title = "CellType")) + 
  theme_ArchR(legendPosition = "bottom", 
              baseSize = 12,
              xText90 = TRUE,
              legendTextSize = 10) + 
  #facet_grid(~ Status) + 
  ggtitle("Majority voting mainly affects cell compositino of Lung 5")
```

### Validating cell type annotations

Marker genes show high gene scores computed from nearby ATAC-seq peaks for the corresponding cluster

```{r message = F, warning = F}
markerGenes  <- c(
    "IL4R", "MS4A1", #B-Cell Trajectory
    "GNLY", "CD8B", "LEF1","TCF7", "IL7R", "CTLA4", "FOXP3" #TCells
  )

p_markers <- plotEmbedding(
    ArchRProj = proj, 
    colorBy = "GeneScoreMatrix", 
    name = markerGenes, 
    embedding = "UMAP",
    quantCut = c(0.01, 0.95),
    imputeWeights = getImputeWeights(proj)
)

p2 <- lapply(p_markers, function(x){
    x + guides(color = FALSE, fill = FALSE) +
    # simplify title 
    ggtitle(strsplit(x$labels$title, ": ")[[1]][2]) + 
    theme_ArchR(baseSize = 6.5) + 
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})

do.call(plot_grid, c(list(ncol = 3), p2))
```



## Annotating clusters with label transferring

Most of the clusters were annotated similarly using both majority voting or label transferring approaches. For the ambiguous clusters, results from label transferring show C16 was mostly Th17 and C14 was mixture of CD8/CD4_T cells. This is due to imperfect clustering or heterogeneity at chromatin levels for cells in those clusters.

```{r, fig.width = 10}
proj@cellColData$LabelTransfer <- 
  factor(metadata$predictedGroup_Un,
         levels = rev(names(pal)))
p1 <- 
  ggplot(proj@cellColData,
         aes(x = Clusters,
             fill = CellType)) + 
    geom_bar() + 
    scale_fill_manual(values = pal) + 
    theme_classic(base_size = 12)  + 
    ggtitle("Cells were partially annotated by matched RNA-seq")
p2 <-   
ggplot(proj@cellColData,
       aes(x = Clusters,
           fill = LabelTransfer)) + 
  geom_bar() + 
  scale_fill_manual(values = pal) + 
  theme_classic(base_size = 12) + 
  ggtitle("Cells were annotated by label transferring")

ggAlignPlots(p1, p2, type = "v")
```

### UMAP for both approaches

```{r, fig.wdith = 12}

proj@cellColData$MajorityVote <-
  factor(proj@cellColData$PredCellType,
         levels = rev(ct_order))

pal_atac <- paletteDiscrete(values = proj@cellColData$MajorityVote)

p1 <- plotEmbedding(
  ArchRProj = proj,
  colorBy = "cellColData",
  name = "LabelTransfer",
  embedding = "UMAP",
  pal = pal
) + theme_ArchR(legendTextSize = 6)


p2 <- plotEmbedding(
  ArchRProj = proj,
  colorBy = "cellColData",
  name = "MajorityVote",
  embedding = "UMAP",
  pal = pal_atac
) + theme_ArchR(legendTextSize = 6)

ggAlignPlots(p1, p2, type = "h")
```

### Track plots for all peaks near TSS of each marker gene 

![](assets/genomic_tracks_marker_genes.png)

```{r message = F, eval = F}
library(locuszoomr)
library(glue)
gene <- c("IL4R", "MS4A1", "GNLY", "CD8B", "LEF1", "CTLA4")
window_size <- 0.1e5
library(EnsDb.Hsapiens.v86)
edb <- EnsDb.Hsapiens.v86

loc_list <-
  lapply(gene, function(i){
    locus(gene = i,
	        flank = window_size,
	        ens_db = edb, )
  })
names(loc_list) <- gene

gene_tracks <-
  lapply(loc_list, function(i){
    gg_genetracks(i, cex.text = 1) + 
#     scale_x_continuous(limits = i$xrange/1e6,
# 		       labels = function(x) round(x, 2),
# 		       expand = c(0, 0)) +
    theme_minimal() + xlab("") + 
    theme(
      axis.text.x = element_blank(),
      plot.background = element_rect(color = "white", fill = "white"))
  }) 

names(gene_tracks) <- gene
# Atac-seq peaks
bigwigs <- "../output/u19_multiomics/analyses/u19_peaks_tissue/GroupBigWigs/CellGroup/" |>
    list.files(full.name = TRUE, pattern = "*.bw")

names(bigwigs) <- 
    unlist(lapply(basename(bigwigs), function(i){strsplit(i, "-")[[1]][1]}))

loc_DF <- data.frame(
  t(sapply(loc_list, function(i){c(i$seqname, i$xrange[1], i$xrange[2])}))) 

plot_list <- list()
for(loc in loc_list){
  interv <- GRanges(glue("chr{loc$seqname}"), IRanges(loc$xrange[1], loc$xrange[2]))

  atac_ranges <- map(bigwigs[1:9], ~import(., which = interv)) 
  
  atac_covered <-
      atac_ranges |>
      map_dfr(~as.data.frame(.) |> as_tibble(), .id = "CellType") |>
      dplyr::select(CellType, start, end, score) %>%
      mutate(CellType = gsub("Lung_", "", CellType))
  
  atac_gaps <- 
      atac_ranges |>
      map_dfr(~ranges(.) |> gaps() |> as.data.frame() |> as_tibble(), .id = "CellType") |>
      mutate(score = 0) |>
      dplyr::select(CellType, start, end, score)
  
  atac_peaks <- 
      bind_rows(atac_covered, atac_gaps) |>
      arrange(CellType, start) |>
      pivot_longer(start:end, names_to = "dummy", values_to = "pos") |>
      dplyr::filter(CellType != "Other") |>
      mutate(CellType = factor(CellType,
                               levels = gsub("/", ".", ct_order),
                               labels = ct_order))
  
  
  atac_plot <- 
      ggplot(atac_peaks) +
      geom_ribbon(aes(x = pos, ymin = 0, ymax = score, color = CellType, fill = CellType),
  		linewidth = .5, outline.type = "full", alpha = .5) +
      scale_x_continuous(
             limits = loc$xrange, 
  		       labels = function(x) round(x/1e6L, 2),
  		       expand = c(0, 0)) +
      scale_color_manual(values = pal_atac) +
      scale_fill_manual(values = pal_atac) +
      facet_wrap(~ CellType, ncol = 1, strip.position = "left") +
      theme_minimal() +
      theme(
  	  legend.position = "none",
  	  axis.text.y = element_blank(),
  	  axis.ticks.y = element_blank(),
  	  #strip.text.y.left = element_text(angle = 0, size = 12),
  	  strip.text.y.left = element_blank(),
  	  panel.grid.major.x = element_blank(),
  	  panel.grid.minor.x = element_blank(),
  	  panel.grid.major.y = element_blank(),
  	  panel.grid.minor.y = element_blank(),
  	  plot.background = element_rect(color = "white", fill = "white")) +
      labs(x = NULL)

    plot_list[[loc$gene]] <- 
        plot_grid(gene_tracks[[loc$gene]], atac_plot, rel_heights  = c(0.2, 1),
                  ncol = 1)
}
```

```{r, eval = F}
ggarrange(plot_list$IL4R,
          plot_list$MS4A1,
          plot_list$GNLY,
          plot_list$CD8B,
          plot_list$LEF1,
          plot_list$CTLA4,
          ncol = 6)
```

