---
title: "testing TF mediating cross-tissue GE"
output: html_document
date: '2025-10-31'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(dplyr)
library(tidyr)
library(ggplot2)
library(data.table)
ct_order <-
  c(
  "Naive_B",
  "Memory_B",
  "NK",
  "CD8_T",
  "CD4_T",
  "Th17",
  "Treg"
)
source("code/make_plots.R")
source("code/utility_function.R")
```

## Cross-tissue DE tested in bulk by DESeq2

100-650 cross-tissue DE genes were found across cell types with single-cell approach. About half amount were detected with bulk approach, except for Memory B cells. 

```{r cars}
load("output/u19_multiomics/output/u19_rna/bulk_cross_tissue_DESeq2.RData")

DEG_tbl <-
  sapply(bulk_DEG[-3:-1], function(i) {
    lung_up = sum(i$p_val_adj <= 0.1 & i$avg_log2FC >0, na.rm = T)
    spleen_up = sum(i$p_val_adj <= 0.1 & i$avg_log2FC <0, na.rm = T)
    return(c(lung_up, spleen_up))})

rownames(DEG_tbl) <- c("lung_up", "spleen_up")
DEG_tbl
```

Top 10 lung-upregulated genes in each cell type

```{r}
# GO enrichment for tissue-specific genes
lung_up <- sapply(bulk_DEG[4:7], function(i) {
   return(
     rownames(
       i %>% filter(p_val_adj <= 0.1 & avg_log2FC > 0) %>%
        slice_max(order_by = avg_log2FC, n = 10)))
})

spleen_up <- sapply(bulk_DEG[4:7], function(i) {
   return(
     rownames(
       i %>% filter(p_val_adj <= 0.1 & avg_log2FC < 0) %>%
        slice_max(order_by = abs(avg_log2FC), n = 10)))
})
```

```{r}
DT::datatable(lung_up, caption = "Lung up-regulated DE genes")
```

```{r}
DT::datatable(spleen_up,
              caption = "Spleen up-regulated DE genes")

```

## Lung-upregulated genes were enriched for GO terms with immune-related functions

Lung-upregulated genes in memory B cells and CD4+ T cells share many pathways related to cell activation and responses to cytokines. These results were similar as previous ones with single-cell approach; however the signals were stronger with bulk approach.

```{r}
library(scales)
go_tbl <- readRDS("output/u19_multiomics/output/GO_enrichment/GRN/lung_upreg_GO_enrichment.RDS")
feature = "Biological"
toshow <- 
    do.call(rbind, go_tbl[-3:-1]) %>% 
      filter(FDR <= 0.05 & grepl(feature, database)) %>%
        group_by(k) %>%
          slice_max(n = 10, order_by = enrichmentRatio) %>%
          pull(description)
p <-
  ggplot(
    do.call(rbind, go_tbl[-3:-1]) %>% 
      filter(FDR <= 0.05 & grepl(feature, database)) %>%
      filter(grepl(feature, database) & description %in% toshow),
    aes(y = description,
        x = factor(gsub("-", "_", k), levels = ct_order),
        size = enrichmentRatio,
        color = -log10(FDR))) +
    # scale_color_gradient(
    #    limits = c(0, 5),
    #    oob = squish) + 
    geom_point() + xlab("") + ylab("") + 
    theme(axis.text.x = element_text(angle = 90))
p
```

## Spleen-upregulated genes were strictly enriched in B cell functions

Somehow spleen-upregulated genes in T cells are enriched for B cell activation. 

```{r}
go_tbl <- readRDS("output/u19_multiomics/output/GO_enrichment/GRN/spleen_upreg_GO_enrichment.RDS")
feature = "Biological"
toshow <- 
    do.call(rbind, go_tbl) %>% 
      filter(FDR <= 0.1 & grepl(feature, database)) %>%
        group_by(k) %>%
          slice_max(n = 15, order_by = enrichmentRatio) %>%
          pull(description)
p <-
  ggplot(
    do.call(rbind, go_tbl) %>% 
      filter(FDR <= 0.1 & grepl(feature, database)) %>%
        filter(grepl(feature, database) & description %in% toshow),
    aes(y = description,
        x = factor(gsub("-", "_", k), levels = ct_order),
        size = enrichmentRatio,
        color = -log10(FDR))) + 
    geom_point() + xlab("") + ylab("") + 
    theme(axis.text.x = element_text(angle = 90))

p
```

In summary, genes with higher expression in lungs are broadly enriched for T cell related functions, while those in spleens are broadly enriched for B cell related functions. We still see lung up-regulated genes enriched in pathways of heat-shock protein genes. Overall, the GO enrichment results are consistent between bulk or single-cell approach.

## Testing whether TFs mediate cross-tissue GE differences

We have shown that chromatin accessibility for cross-tissue DE genes are very similar. Thus, some other mechanisms may drive the differences, which motivates us to test whether TF activity is responsible for the up-regulation of genes in lung. 

Cell composition for both tissues
![cell comosition](assets/rna_cell_composition.png)

We examined GRNs for B cells because both tissues have relatively similar number of cells. Though half amount of memory B cells found in lung compared to spleen, we still detected several lung TFs with higher activity relative to spleen. The differences in naive B cells are small, which is consistent with lack of differences in cross-tissue DE genes in naive B cells.  

```{r}
ct_order <-
  c(
  "Naive_B",
  "Memory_B",
  "CD8_T",
  "CD4_T",
  "Th17",
  "Treg"
)
cutoffs <- c(0.01, 0.015, 0.02, 0.025, 0.03, 0.035)
target_list <-list()
all_list <- list()
for(c in cutoffs){
  print(c)
  lung_grn <-
    lapply(ct_order, function(i) {
      f <- fread(sprintf("analysis/dictys/sparsity%s/L_%s_GRN.txt", c, i))
      return(lapply(split(f, f$Regulator), function(i){i$Target}))
    })
  names(lung_grn) <- ct_order
  target_list[[as.character(c)]] <- lung_grn
  all_genes <-
     lapply(ct_order, function(i) {
      f <- fread(sprintf("analysis/dictys/sparsity%s/L_%s_GRN.txt", c, i))
      return(unique(f$Target))
    })
  names(all_genes) <- ct_order
  all_list[[as.character(c)]] <- all_genes
}
```


```{r, fig.width = 5, fig.height = 5}
library(ggrepel)

plots <-lapply(c("Naive_B", "Memory_B"), function(i){
  f <- fread(sprintf("analysis/dictys/sparsity0.02/%s_cross_tissue_DE_DR_logFC.txt", i))
  colnames(f)[1] <- "gene"
    ggplot(
      f, 
      aes(x = DE_logFC,
          y = DR_logFC,
          label = ifelse(DE_logFC >= 1 & DR_logFC >= 1 |
                           (DE_logFC <= -1 & DR_logFC <= -1), gene, ""))) + 
      geom_point(size = 0.5) + 
      geom_text_repel(size = 4) + 
      geom_vline(xintercept = 0) + 
      geom_hline(yintercept = 0) + 
      theme_pub() +  
      ggtitle(sprintf("Lung vs. Spleen: %s", i))
})

plots
```
Let's focus on TFs specific to lung memory B cells and ask whether they explain up-regulated genes in lungs

```{r}
btw_tissue <- readRDS("output/u19_multiomics/output/DE_analysis_wilcox/cross_tissue_DE_wilcox.RDS")

DGE_tbl <- do.call(rbind, btw_tissue)
DGE_tbl$cluster <- unlist(lapply(rownames(DGE_tbl), function(i){strsplit(i, "[.]")[[1]][1]}))
DGE_tbl$gene <- unlist(lapply(rownames(DGE_tbl), function(i){strsplit(i, "[.]")[[1]][2]}))

sc_lungUp <-
    DGE_tbl %>% filter(cluster == "Memory_B" & 
                         p_val_adj <= 0.05 &
                         avg_log2FC >= 0.58) 
bulk_lungUp <- bulk_DEG$`Memory-B` %>% 
  mutate(gene = rownames(bulk_DEG$`Memory-B`)) %>%
  filter(p_val_adj <= 0.05 & avg_log2FC >= 0.58)

f <- fread(sprintf("analysis/dictys/sparsity0.02/%s_cross_tissue_DE_DR_logFC.txt", "Memory_B"))
regTFs <- f %>% filter(DE_logFC >= 0.58)
colnames(regTFs)[1] <- "gene"
```


### Testing whether lung-specific TFs regulated more DE genes

Here I performed the tests on DE genes identified using single-cell or bulk approach. I identified lung-specific TFs by only requiring its expression to be 1.5 fold higher in lung than spleen. I relaxed the differential regulation criteria because the quantity of target counts do not truly reflect the difference. 
```{r}
# enrich_out <-
#   lapply(list(sc_lungUp, bulk_lungUp), function(i) {
#     EnrichmentTest(var1 = regTFs$gene,
#                  var2 = i$gene,
#                  bkg = names(target_list$`0.02`$Memory_B))[[2]]
#   })
# 
# enrich_out <- data.frame(do.call(rbind, enrich_out))
# rownames(enrich_out) <- c("single-cell_DE", "bulk_DE")
# enrich_out
# print("The list of memory B TFs that have higher expression in lung:")
# overlapped_tfs <-
#   lapply(list(sc_lungUp, bulk_lungUp), function(i) {
#     intersect(i$gene, regTFs$gene)
#   })
# names(overlapped_tfs) <- c("single-cell_DE", "bulk_DE")
# overlapped_tfs

# Target genes of more enriched for lung-up genes?

sc_out <-
  lapply(regTFs$gene, function(i) {
    target <- unique(target_list$`0.02`$Memory_B[[i]])
    out_tbl = EnrichmentTest(target, sc_lungUp$gene, all_list$`0.02`$Memory_B)
    return(out_tbl[[2]])
  })
sc_out <- data.frame(do.call(rbind, sc_out)) %>%
  mutate(FDR = p.adjust(p_value, method = "fdr"))
rownames(sc_out) <- regTFs$gene

bulk_out <-
  lapply(regTFs$gene, function(i) {
    target <- unique(target_list$`0.02`$Memory_B[[i]])
    out_tbl = EnrichmentTest(target, bulk_lungUp$gene, all_list$`0.02`$Memory_B)
    return(out_tbl[[2]])
  })

bulk_out <- data.frame(do.call(rbind, bulk_out)) %>%
  mutate(FDR = p.adjust(p_value, method = "fdr"))
rownames(bulk_out) <- regTFs$gene

print("Results for single-cell approach DE genes:")
sc_out %>% 
  filter(FDR <= 0.1 & odds_ratio > 1)

print("Results for bulk approach DE genes:")
bulk_out %>%
  filter(FDR <= 0.3 & odds_ratio > 1)
```

### Examining the effect sizes between TF regulation and differential expression

As we identified lung-specific TFs enriched for DE genes, we further asked if TFs that have stronger links tend to have DE genes with larger differences between lung and spleen. Let's use gene expression in spleen as baseline. If specific lung TFs drive gene expression in lung to be higher than baseline, we would expect to higher TF activity leads to higher expression. 

**Method**
For each lung-specific TF, we identified lung up-regulated genes in its network. Then we plotted TF strength and differential expression and showed the fitted line.


Bulk DE genes

```{r}
f <- fread("analysis/dictys/sparsity0.02/L_Memory_B_GRN.txt")
lapply(rownames(bulk_out[bulk_out$FDR <= 0.3 & bulk_out$odds_ratio > 1, ]), function(i) {
  common <- intersect(f$Target[f$Regulator == i], bulk_lungUp$gene)
  df <-
    data.frame(DR = f$Strength[f$Regulator == i & f$Target %in% common],
               DE = bulk_lungUp$avg_log2FC[bulk_lungUp$gene %in% common])

  model = lm(df$DE ~ df$DR)
  #out_tbl = rbind(out_tbl, c(i, round(coef(summary(model))[2, ], 2)))
  coeff = as.numeric(coefficients(model))
  plot(x = df$DR, y = df$DE,
       ylab="DE_logFC", 
       xlab="Regulation strength", cex = 0.8,
       main = i, cex.axis = 1.5,
       pch = 20)
  abline(a = coeff[1], b = coeff[2])
})


```
Single-cell DE genes

```{r}
lapply(rownames(sc_out[sc_out$FDR <= 0.1 & sc_out$odds_ratio > 1, ]), function(i) {
  common <- intersect(f$Target[f$Regulator == i], sc_lungUp$gene)
  df <-
    data.frame(DR = f$Strength[f$Regulator == i & f$Target %in% common],
               DE = sc_lungUp$avg_log2FC[sc_lungUp$gene %in% common])
  model = lm(df$DE ~ df$DR)
  #out_tbl = rbind(out_tbl, c(i, round(coef(summary(model))[2, ], 2)))
  coeff = as.numeric(coefficients(model))
  plot(x = df$DR, y = df$DE,
       ylab="DE_logFC", 
       xlab="Regulation strength", cex = 0.8,
       main = i, cex.axis = 1.5,
       pch = 20)
  abline(a = coeff[1], b = coeff[2])
})
```
We saw positive trends for the two TFs enriched for bulk-level DE genes, but much weaker trend for the three TFs enriched for single-cell level DE genes. 
